<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="occumb">
<title>Model specification in occumb() • occumb</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Model specification in occumb()">
<meta property="og:description" content="occumb">
<meta name="robots" content="noindex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">occumb</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.0.2.9017</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/occumb.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/model_specification.html">Model specification in occumb()</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/fukayak/occumb/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Model specification in occumb()</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/fukayak/occumb/blob/HEAD/vignettes/model_specification.Rmd" class="external-link"><code>vignettes/model_specification.Rmd</code></a></small>
      <div class="d-none name"><code>model_specification.Rmd</code></div>
    </div>

    
    
<p>The <code><a href="../reference/occumb.html">occumb()</a></code> function implements a hierarchical model
that represents sequence read count data obtained from spatially
replicated environmental DNA metabarcoding as a consequence of
sequential stochastic processes (i.e., ecological and observational
processes). The model allows us to account for false-negative species
detection errors that occur at different stages of the metabarcoding
workflow under several assumptions, including that there are no false
positive errors. To identify model parameters replicates at different
levels are required; i.e., you should have multiple sites and multiple
within-site replicates. However, it is not necessary to have a balanced
design with the same number of replicates at all sites. A quick and less
formal description of this model is provided in the following. Readers
are encouraged to refer to the <a href="https://doi.org/10.1111/2041-210X.13732" class="external-link">original paper</a> for a
more formal and complete explanation.</p>
<div class="section level2">
<h2 id="modeling-the-sequential-sampling-process-of-edna-metabarcoding">Modeling the sequential sampling process of eDNA metabarcoding<a class="anchor" aria-label="anchor" href="#modeling-the-sequential-sampling-process-of-edna-metabarcoding"></a>
</h2>
<p>We assume that the occurrence of <code>I</code> focal species was
monitored at <code>J</code> sites sampled from an area of interest. At
site <code>j</code>, <code>K[j]</code> replicates of environmental
samples are collected. For each replicate, a library for DNA sequencing
is prepared to obtain separate sequence reads for each replicate. We
denote the resulting sequence read counts of species <code>i</code> for
replicate <code>k</code> at site <code>j</code>, obtained using
high-throughput sequencing and subsequent bioinformatic processing, as
<code>y[i, j, k]</code>.</p>
<div class="figure">
<img src="images/model.png" alt=""><p class="caption">Adapted from Figure 1 in Fukaya et al. 2022 (CC BY
4.0)</p>
</div>
<p>The figure above shows a diagram of the minimal model that can be
fitted by <code><a href="../reference/occumb.html">occumb()</a></code> with default settings. The process of
generating <code>y</code> is represented by a series of latent variables
<code>z</code>, <code>u</code>, and <code>r</code>, and parameters
<code>psi</code>, <code>theta</code>, and <code>phi</code> that govern
the variation of the latent variables. Although <code>psi</code>,
<code>theta</code>, and <code>phi</code> are assumed here to have
species-specific values, modeling these parameters as functions of
covariates allows for further variation in them (see the following
section).</p>
<p>First, look at the right-pointing arrow on the left, representing the
ecological process of species distribution. For the DNA sequence of
species to be detected at a site, the site must be occupied by the
species (i.e., the species’ environmental DNA must be present at the
site). In the model, the site occupancy of species is indicated by the
latent variable <code>z[i, j]</code>. If site <code>j</code> is occupied
by species <code>i</code>, then <code>z[i, j] = 1</code>; otherwise,
<code>z[i, j] = 0</code>. <code>psi[i]</code> represents the probability
that species <code>i</code> occupies a site selected from the region of
interest. Thus, species with high <code>psi[i]</code> values should
occur at many sites in the region, while species with low
<code>psi[i]</code> values should occur at a limited number of
sites.</p>
<p>Next, focus on the second right-pointing arrow from the left, which
represents one of the two observation processes, the capture of species
DNA sequence. For the DNA sequence of species to be detected, the
environmental sample collected at an occupied site (i.e.,
<code>z[i, j] = 1</code>) must contain species eDNA and the amplicon
derived from it must be present in the prepared sequencing library. In
the model, the inclusion of the DNA sequence of species in a sequencing
library is indicated by the latent variable <code>u[i, j, k]</code>. If
the library of replicate <code>k</code> at site <code>j</code> contains
the DNA sequence of species <code>i</code>, then
<code>u[i, j, k] = 1</code>; otherwise, <code>u[i, j, k] = 0</code>.
<code>theta[i]</code> represents the probability that the DNA sequence
of species <code>i</code> is captured per replicate collected at a site
occupied by the species. DNA sequences of species with high
<code>theta[i]</code> values are more reliably captured at occupied
sites, while DNA sequences of species with low <code>theta[i]</code>
values are more difficult to capture. Note that <code>u[i, j, k]</code>
always takes zero for sites not occupied by species (i.e.,
<code>z[i, j] = 0</code>), assuming that false positives are not to
occur.</p>
<p>Finally, look at the right-pointing arrow on the right side. This
model part represents another observation process, the allocation of
species sequence reads in high-throughput sequencing. Here, the sequence
read count vector <code>y[1:I, j, k]</code> is assumed to follow a
multinomial distribution with the total number of sequence reads in
replicate <code>k</code> of site <code>j</code> as the number of trials.
Its multinomial cell probability, <code>pi[1:I, j, k]</code> (not shown
in the figure), is modeled as a function of latent variables
<code>u[1:I, j, k]</code> described above and <code>r[1:I, j, k]</code>
that is proportional to the relative frequency of species sequence
reads. The variation of <code>r[i, j, k]</code> is governed by the
parameter <code>phi[i]</code>, which represents the relative dominance
of species sequence. Species with higher <code>phi[i]</code> values tend
to have relatively more reads when the species sequence is included in
the library (i.e. <code>u[i, j, k] = 1</code>), while species with lower
<code>phi[i]</code> values tend to have relatively fewer reads. No false
positives are assumed to occur at this stage, too; that is,
<code>pi[i, j, k]</code> always takes zero for replicates that do not
include the species DNA sequence (i.e.,
<code>u[i, j, k] = 0</code>).</p>
<p>In the figure, the arrows directed at <code>psi[i]</code>,
<code>theta[i]</code>, and <code>phi[i]</code> indicate that the
variation of these parameters is governed by a community-level
multivariate normal prior distribution with mean <code>Mu</code> and
covariance matrix <code>Sigma</code>. Details are discussed below.</p>
</div>
<div class="section level2">
<h2 id="covariate-modeling-of-psi-theta-and-phi">Covariate modeling of <code>psi</code>, <code>theta</code>, and
<code>phi</code><a class="anchor" aria-label="anchor" href="#covariate-modeling-of-psi-theta-and-phi"></a>
</h2>
<p>Variation in <code>psi</code>, <code>theta</code>, and
<code>phi</code> can be modeled as a function of covariates in a manner
similar to generalized linear models (GLMs) and generalized linear mixed
models (GLMMs). That is, the covariates are incorporated into linear
predictors on the appropriate link scales for the parameters (logit for
<code>psi</code> and <code>theta</code> and log for <code>phi</code>).
The <code><a href="../reference/occumb.html">occumb()</a></code> function allows for covariate modeling using
standard R formula syntax.</p>
<p>There are three types of related covariates: <strong>species
covariates</strong> that can take on different values for each species
(e.g., traits), <strong>site covariates</strong> that can take on
different values for each site (e.g., environment), and
<strong>replicate covariates</strong> that can take on different values
for each combination of site and replicate (e.g., the amount of water
filtered). These covariates can be included in the data object via the
<code>spec_cov</code>, <code>site_cov</code>, and <code>repl_cov</code>
arguments of the <code><a href="../reference/occumbData.html">occumbData()</a></code> function and used to specify
models in the <code><a href="../reference/occumb.html">occumb()</a></code> function.</p>
<p>The <code><a href="../reference/occumb.html">occumb()</a></code> function specifies covariates for each
parameter using the <code>formula_&lt;parameter name&gt;</code> and
<code>formula_&lt;parameter name&gt;_shared</code> arguments. The
<code>formula_&lt;parameter name&gt;</code> and
<code>formula_&lt;parameter name&gt;_shared</code> arguments are used to
specify species-specific effects and effects shared by all species,
respectively. The following table shows examples of modeling
<code>psi</code> using the <code>formula_psi</code> and
<code>formula_psi_shared</code> arguments where <code>i</code> is the
species index, <code>j</code> is the site index, <code>speccov1</code>
is a continuous species covariate, and <code>sitecov1</code> and
<code>sitecov2</code> are continuous site covariates. <code>psi</code>
can be modeled as a function of species and site covariates.</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th align="left"><code>formula_psi</code></th>
<th align="left"><code>formula_psi_shared</code></th>
<th align="left">Linear predictor specified</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(psi[i]) = gamma[i, 1]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ sitecov1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(psi[i, j]) = gamma[i, 1] + gamma[i, 2] * sitecov1[j]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ sitecov1 + sitecov2</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(psi[i, j]) = gamma[i, 1] + gamma[i, 2] * sitecov1[j] + gamma[i, 3] * sitecov2[j]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ sitecov1 * sitecov2</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(psi[i, j]) = gamma[i, 1] + gamma[i, 2] * sitecov1[j] + gamma[i, 3] * sitecov2[j] + gamma[i, 4] * sitecov1[j] * sitecov2[j]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ speccov1</code></td>
<td align="left"><code>logit(psi[i]) = gamma[i, 1] + gamma_shared[1] * speccov1[i]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ sitecov1</code></td>
<td align="left"><code>logit(psi[i, j]) = gamma[i, 1] + gamma_shared[1] * sitecov1[j]</code></td>
</tr>
</tbody>
</table>
<p>In <code><a href="../reference/occumb.html">occumb()</a></code>, species-specific effects on
<code>psi</code> are denoted as <code>gamma</code>, and shared effects
on <code>psi</code> are denoted as <code>gamma_shared</code>. The first
case specifies a default, intercept-only model.
<code>logit(psi[i])</code> is therefore determined only by the intercept
term <code>gamma[i, 1]</code>. Note that <code><a href="../reference/occumb.html">occumb()</a></code> always
estimates the species-specific intercept <code>gamma[i, 1]</code>, as in
this simplest case. In the second case, species-specific effects
<code>gamma[i, 2]</code> of the site covariate <code>sitecov1</code> are
incorporated. Note here that the site subscript <code>j</code> is added
to <code>psi</code> on the left-hand side of the equation because the
value of <code>psi</code> now varies from site to site depending on the
value of <code>sitecov1[j]</code>. In the third and fourth cases,
another site covariate, <code>sitecov2</code>, is specified in addition
to <code>sitecov1</code>. As shown in the fourth case, interaction terms
can be specified using the <code>*</code> operator.</p>
<p>In the fifth case, the <code>formula_psi_shared</code> argument
specifies the shared effect of the species covariate
<code>speccov1</code>. Note that the effect <code>gamma_shared[1]</code>
of <code>speccov1[i]</code> in the linear predictor does not have
subscript <code>i</code>. Because species-specific effects cannot be
estimated for species covariates, <code><a href="../reference/occumb.html">occumb()</a></code> accepts species
covariates and their interactions only in its
<code>formula_&lt;parameter name&gt;_shared</code> argument. Introducing
species covariates does not change the dimension of <code>psi</code>
(note that it has only the subscript <code>i</code>) but may help reveal
variations in site occupancy probability associated with species
characteristics.</p>
<p>In the sixth case, the site covariate <code>sitecov1</code> is
specified in the <code>formula_psi_shared</code> argument. In contrast
to the second case, <code>sitecov1[j]</code> has a shared effect
<code>gamma_shared[1]</code> in the linear predictor. Since species are
often expected to respond differently to site characteristics, site
covariates are likely to be introduced using the
<code>formula_psi</code> argument. Nevertheless, the
<code>formula_psi_shared</code> argument can be used when assuming
consistent covariate effects across species is reasonable or when the
data support doing so.</p>
<p>A similar approach can be applied to <code>theta</code> and
<code>phi</code>, which can be modeled as a function of species, site,
and replicate covariates. The following table shows examples of modeling
for <code>theta</code> where <code>i</code> is the species index,
<code>j</code> is the site index, <code>k</code> is the replicate index,
<code>speccov1</code> is a continuous species covariate,
<code>sitecov1</code> is a continuous site covariate, and
<code>replcov1</code> is a continuous replicate covariate.</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th align="left"><code>formula_theta</code></th>
<th align="left"><code>formula_theta_shared</code></th>
<th align="left">Linear predictor specified</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(theta[i]) = beta[i, 1]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ sitecov1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(theta[i, j]) = beta[i, 1] + beta[i, 2] * sitecov1[j]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ replcov1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(theta[i, j, k]) = beta[i, 1] + beta[i, 2] * replcov1[j, k]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ speccov1</code></td>
<td align="left"><code>logit(theta[i]) = beta[i, 1] + beta_shared[1] * speccov1[i]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ sitecov1</code></td>
<td align="left"><code>logit(theta[i, j]) = beta[i, 1] + beta_shared[1] * sitecov1[j]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ replcov1</code></td>
<td align="left"><code>logit(theta[i, j, k]) = beta[i, 1] + beta_shared[1] * replcov1[j, k]</code></td>
</tr>
</tbody>
</table>
<p>In <code><a href="../reference/occumb.html">occumb()</a></code>, species-specific effects on
<code>theta</code> are denoted as <code>beta</code>, and shared effects
on <code>theta</code> are denoted as <code>beta_shared</code>. The first
case specifies an intercept-only model. As in the case of
<code>psi</code>, <code><a href="../reference/occumb.html">occumb()</a></code> always estimates the
species-specific intercept <code>beta[i, 1]</code>. The second and third
cases can be contrasted with the second case of the <code>psi</code>
example with a single covariate specified in the
<code>formula_psi</code> argument, and the remaining other cases with
the fifth and sixth cases of the <code>psi</code> example with a single
covariate specified in the <code>formula_psi_shared</code> argument.
Because replicate covariate <code>replcov1</code> have both the site
index <code>j</code> and replicate index <code>k</code>, specifying it
adds these two indexes to <code>theta</code>.</p>
<p>The same rule applies to <code>phi</code>. Here is an example of more
complex cases involving interactions between different types of
covariates.</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th align="left"><code>formula_phi</code></th>
<th align="left"><code>formula_phi_shared</code></th>
<th align="left">Linear predictor specified</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>log(phi[i]) = alpha[i, 1]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ sitecov1 * replcov1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>log(phi[i, j, k]) = alpha[i, 1] + alpha[i, 2] * sitecov1[j] + alpha[i, 3] * replcov1[j, k] + alpha[i, 4] * sitecov1[j] * replcov1[j, k]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ replcov1</code></td>
<td align="left"><code>~ speccov1 * sitecov1</code></td>
<td align="left"><code>log(phi[i, j, k]) = alpha[i, 1] + alpha[i, 2] * replcov1[j, k] + alpha_shared[1] * speccov1[i] + alpha_shared[2] * sitecov1[j] + alpha_shared[3] * speccov1[i] * sitecov1[j]</code></td>
</tr>
</tbody>
</table>
<p>In <code><a href="../reference/occumb.html">occumb()</a></code>, species-specific effects on
<code>phi</code> are denoted as <code>alpha</code>, and shared effects
on <code>phi</code> are denoted as <code>alpha_shared</code>. As with
the other two parameters, <code><a href="../reference/occumb.html">occumb()</a></code> always estimates the
species-specific intercept <code>alpha[i, 1]</code>.</p>
<p>The following table summarizes the covariate types accepted by each
<code>formula</code> argument.</p>
<table class="table">
<thead><tr class="header">
<th align="left">Argument</th>
<th align="center"><code>spec_cov</code></th>
<th align="center"><code>site_cov</code></th>
<th align="center"><code>repl_cov</code></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>formula_phi</code></td>
<td align="center"></td>
<td align="center">✓</td>
<td align="center">✓</td>
</tr>
<tr class="even">
<td align="left"><code>formula_theta</code></td>
<td align="center"></td>
<td align="center">✓</td>
<td align="center">✓</td>
</tr>
<tr class="odd">
<td align="left"><code>formula_psi</code></td>
<td align="center"></td>
<td align="center">✓</td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="left"><code>formula_phi_shared</code></td>
<td align="center">✓</td>
<td align="center">✓</td>
<td align="center">✓</td>
</tr>
<tr class="odd">
<td align="left"><code>formula_theta_shared</code></td>
<td align="center">✓</td>
<td align="center">✓</td>
<td align="center">✓</td>
</tr>
<tr class="even">
<td align="left"><code>formula_psi_shared</code></td>
<td align="center">✓</td>
<td align="center">✓</td>
<td align="center"></td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="prior-distributions">Prior distributions<a class="anchor" aria-label="anchor" href="#prior-distributions"></a>
</h2>
<p>A hierarchical prior distribution is specified for the
species-specific effects <code>alpha</code>, <code>beta</code>, and
<code>gamma</code>. Specifically, the vector of these effects is assumed
to follow a multivariate normal distribution, and a prior distribution
is specified for the elements of its mean vector <code>Mu</code> and
covariance matrix <code>Sigma</code> whose values are estimated from the
data. Because these hyperparameters summarize the variation in
species-specific effects at the community level, their estimates may be
of interest.</p>
<p>A normal prior distribution with mean 0 and precision
<code>prior_prec</code> is specified for each element of
<code>Mu</code>. The <code>prior_prec</code> value is determined by the
<code>prior_prec</code> argument of the <code><a href="../reference/occumb.html">occumb()</a></code> function,
which by default is set to a small value <code>1e-4</code> to specify
vague priors. Note that precision is the inverse of variance.</p>
<p><code>Sigma</code> is decomposed into elements of standard deviation
<code>sigma</code> and correlation coefficient <code>rho</code>, each
specified with a different vague prior. Specifically, a uniform prior
distribution with a lower limit of 0 and an upper limit of
<code>prior_ulim</code> is specified for <code>sigma</code>, and a
uniform prior with a lower limit of −1 and an upper limit of 1 is set
for <code>rho</code>. The value of <code>prior_ulim</code> is determined
by the <code>prior_ulim</code> argument of the <code><a href="../reference/occumb.html">occumb()</a></code>
function and is set to <code>1e4</code> by default.</p>
<p>For each of the shared effects <code>alpha_shared</code>,
<code>beta_shared</code>, and <code>gamma_shared</code>, a normal prior
distribution with mean 0 and precision <code>prior_prec</code> is
specified.</p>
</div>
<div class="section level2">
<h2 id="after-all-which-are-the-relevant-parameters">After all, which are the relevant parameters?<a class="anchor" aria-label="anchor" href="#after-all-which-are-the-relevant-parameters"></a>
</h2>
<p>The latent variables and parameters of the model to be estimated and
saved using the <code><a href="../reference/occumb.html">occumb()</a></code> function is as follows (note that
<code><a href="../reference/occumb.html">occumb()</a></code> will not save <code>u</code> and <code>r</code>,
but their function <code>pi</code>). Posterior samples of these latent
variables and parameters can be accessed using
<code><a href="../reference/get_posterior.html">get_post_samples()</a></code> or <code><a href="../reference/get_posterior.html">get_post_summary()</a></code>
functions.</p>
<dl>
<dt>
<code>z</code>
</dt>
<dd>
Site occupancy status of species.
</dd>
<dt>
<code>pi</code>
</dt>
<dd>
Multinomial probabilities of species sequence read counts.
</dd>
<dt>
<code>phi</code>
</dt>
<dd>
Sequence relative dominance of species.
</dd>
<dt>
<code>theta</code>
</dt>
<dd>
Sequence capture probabilities of species.
</dd>
<dt>
<code>psi</code>
</dt>
<dd>
Site occupancy probabilities of species.
</dd>
<dt>
<code>alpha</code>
</dt>
<dd>
Species-specific effects on sequence relative dominance
(<code>phi</code>).
</dd>
<dt>
<code>beta</code>
</dt>
<dd>
Species-specific effects on sequence capture probabilities
(<code>theta</code>).
</dd>
<dt>
<code>gamma</code>
</dt>
<dd>
Species-specific effects on site occupancy probabilities
(<code>psi</code>).
</dd>
<dt>
<code>alpha_shared</code>
</dt>
<dd>
Effects on sequence relative dominance (<code>phi</code>) common across
species.
</dd>
<dt>
<code>beta_shared</code>
</dt>
<dd>
Effects on sequence capture probabilities (<code>theta</code>) that are
common across species.
</dd>
<dt>
<code>gamma_shared</code>
</dt>
<dd>
Effects on site occupancy probabilities (<code>psi</code>) that are
common across species.
</dd>
<dt>
<code>Mu</code>
</dt>
<dd>
Community-level averages of the species-specific effects
(<code>alpha</code>, <code>beta</code>, <code>gamma</code>).
</dd>
<dt>
<code>sigma</code>
</dt>
<dd>
Standard deviations of the species-specific effects (<code>alpha</code>,
<code>beta</code>, <code>gamma</code>).
</dd>
<dt>
<code>rho</code>
</dt>
<dd>
Correlation coefficients of the species-specific effects
(<code>alpha</code>, <code>beta</code>, <code>gamma</code>).
</dd>
</dl>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Keiichi Fukaya.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
