<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Model specification in occumb() • occumb</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Model specification in occumb()">
<meta name="robots" content="noindex">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">occumb</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/occumb.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/model_specification.html">Model specification in occumb()</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/fukayak/occumb/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Model specification in occumb()</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/fukayak/occumb/blob/develop/vignettes/model_specification.Rmd" class="external-link"><code>vignettes/model_specification.Rmd</code></a></small>
      <div class="d-none name"><code>model_specification.Rmd</code></div>
    </div>

    
    
<p>The <code><a href="../reference/occumb.html">occumb()</a></code> function implements a hierarchical model
that represents the sequence read count data obtained from spatially
replicated eDNA metabarcoding as a consequence of sequential stochastic
processes (i.e., ecological and observational processes). This model
allows us to account for false-negative species detection errors that
occur at different stages of the metabarcoding workflow. To identify the
model parameters, replicates at different levels are required; that is,
one should have multiple sites and multiple within-site replicates.
However, it is not necessary to have a balanced design with the same
number of replicates at all sites. A brief and less formal description
of this model is provided below. Readers are encouraged to refer to the
<a href="https://doi.org/10.1111/2041-210X.13732" class="external-link">original paper</a> for
a more formal and complete explanation.</p>
<div class="section level2">
<h2 id="modeling-the-sequential-sampling-process-of-edna-metabarcoding">Modeling the sequential sampling process of eDNA metabarcoding<a class="anchor" aria-label="anchor" href="#modeling-the-sequential-sampling-process-of-edna-metabarcoding"></a>
</h2>
<p>We assumed that there were <code>I</code> focal species to be
monitored and <code>J</code> sites sampled from an area of interest. At
site <code>j</code>, <code>K[j]</code> replicates of environmental
samples were collected. For each replicate, a library was prepared for
DNA sequencing to obtain separate sequence reads. We denote the
resulting sequence read count of species <code>i</code> for replicate
<code>k</code> at site <code>j</code> obtained using high-throughput
sequencing and subsequent bioinformatics processing as
<code>y[i, j, k]</code> (<code>i = 1, 2, ..., I</code>;
<code>j = 1, 2, ..., J</code>; <code>k = 1, 2, ..., K[j]</code>).</p>
<div class="float">
<img src="images/model.png" alt="Adapted from Figure 1 in Fukaya et al. 2022 (CC BY 4.0)"><div class="figcaption">Adapted from Figure 1 in Fukaya et al. 2022 (CC
BY 4.0)</div>
</div>
<p>The figure above shows a diagram of the minimal model that can be
fitted using <code><a href="../reference/occumb.html">occumb()</a></code> with default settings. The process of
generating <code>y</code> is represented by a series of latent variables
<code>z</code>, <code>u</code>, and <code>r</code>, and the parameters
<code>psi</code>, <code>theta</code>, and <code>phi</code> that govern
the variation of the latent variables. Although <code>psi</code>,
<code>theta</code>, and <code>phi</code> are assumed to have
species-specific values, modeling these parameters as functions of
covariates allows for further variation (see the following section).</p>
<p>The right-pointing arrow on the left-hand side of the figure
represents the ecological process of species distribution. For the DNA
sequence of a species to be detected at a site, the site must be
occupied by that species (i.e., the eDNA of the species must be present
at the site). In this model, the site occupancy of a species is
indicated by the latent variable <code>z[i, j]</code>. If site
<code>j</code> is occupied by species <code>i</code>,
<code>z[i, j] = 1</code>; otherwise, <code>z[i, j] = 0</code>.
<code>psi[i]</code> represents the probability that species
<code>i</code> occupies a site selected from the region of interest.
Therefore, species with high <code>psi[i]</code> values should occur at
many sites in the region, whereas those with low <code>psi[i]</code>
values should occur at only a limited number of sites.</p>
<p>Next, we focus on the second right-pointing arrow from the left,
which represents one of the two observation processes: the capture of
the species DNA sequence. For the DNA sequence of a species to be
detected, the environmental sample from an occupied site and sequencing
library derived from it must contain the species DNA sequence. In this
model, the inclusion of the DNA sequence of a species in a sequencing
library is indicated by latent variable <code>u[i, j, k]</code>. If the
library of replicate <code>k</code> at site <code>j</code> contains the
DNA sequence of species <code>i</code>, <code>u[i, j, k] = 1</code>;
otherwise, <code>u[i, j, k] = 0</code>. <code>theta[i]</code> represents
the per-replicate probability that the DNA sequence of species
<code>i</code> is captured at a site occupied by the species. The DNA
sequences of species with high <code>theta[i]</code> values are more
reliably captured at occupied sites, whereas those of species with low
<code>theta[i]</code> values are more difficult to capture. Note that
<code>u[i, j, k]</code> is always zero for sites not occupied by a
species (i.e., <code>z[i, j] = 0</code>), assuming that false positives
do not occur.</p>
<p>Finally, we examine the right-pointing arrow on the right-hand side
of the figure. This model part represents another observation process,
that is, the allocation of species sequence reads in high-throughput
sequencing. The sequence read count vector <code>y[1:I, j, k]</code> is
assumed to follow a multinomial distribution, with the total number of
sequence reads in replicate <code>k</code> of site <code>j</code> as the
number of trials. Its multinomial cell probability,
<code>pi[1:I, j, k]</code> (not shown in the figure), is modeled as a
function of the latent variables <code>u[1:I, j, k]</code> described
above and <code>r[1:I, j, k]</code>, which is proportional to the
relative frequency of the species sequence reads. The variation in
<code>r[i, j, k]</code> is governed by parameter <code>phi[i]</code>,
which represents the relative dominance of a species sequence. Species
with higher <code>phi[i]</code> values tend to have more reads when the
species sequence was included in the library
(<code>u[i, j, k] = 1</code>), whereas species with lower
<code>phi[i]</code> values tend to have fewer reads. Again, no false
positives are assumed to occur at this stage; that is,
<code>pi[i, j, k]</code> always takes zero for replicates that do not
include the species DNA sequence (i.e.,
<code>u[i, j, k] = 0</code>).</p>
<p>In the figure, the arrows directed at <code>psi[i]</code>,
<code>theta[i]</code>, and <code>phi[i]</code> indicate that the
variation in these parameters is governed by a community-level
multivariate normal prior distribution with a mean vector
<code>Mu</code> and covariance matrix <code>Sigma</code>. The two
components of <code>Sigma</code> are the standard deviation
<code>sigma</code> and the correlation coefficient <code>rho</code> (not
shown in the figure).</p>
</div>
<div class="section level2">
<h2 id="covariate-modeling-of-psi-theta-and-phi">Covariate modeling of <code>psi</code>, <code>theta</code>, and
<code>phi</code><a class="anchor" aria-label="anchor" href="#covariate-modeling-of-psi-theta-and-phi"></a>
</h2>
<p>Variations in <code>psi</code>, <code>theta</code>, and
<code>phi</code> can be modeled as functions of covariates in a manner
similar to generalized linear models (GLMs). That is, the covariates are
incorporated into linear predictors on the appropriate link scales for
the parameters (logit for <code>psi</code> and <code>theta</code>, and
log for <code>phi</code>). The <code><a href="../reference/occumb.html">occumb()</a></code> function allows
covariate modeling using the standard R formula syntax.</p>
<p>There are three types of related covariates: <strong>species
covariates</strong> that can take on different values for each species
(e.g., traits), <strong>site covariates</strong> that can take on
different values for each site (e.g., habitat characteristics), and
<strong>replicate covariates</strong> that can take on different values
for each replicate (e.g., amount of water filtered). These covariates
can be included in the data object via the <code>spec_cov</code>,
<code>site_cov</code>, and <code>repl_cov</code> arguments of the
<code><a href="../reference/occumbData.html">occumbData()</a></code> function and used to specify the models in the
<code><a href="../reference/occumb.html">occumb()</a></code> function.</p>
<p>The <code><a href="../reference/occumb.html">occumb()</a></code> function specifies covariates for each
parameter using the <code>formula_&lt;parameter name&gt;</code> and
<code>formula_&lt;parameter name&gt;_shared</code> arguments. The
<code>formula_&lt;parameter name&gt;</code> and
<code>formula_&lt;parameter name&gt;_shared</code> arguments are used to
specify species-specific effects and effects shared by all species,
respectively. The following table shows examples of modeling
<code>psi</code> using the <code>formula_psi</code> and
<code>formula_psi_shared</code> arguments, where <code>i</code> is the
species index, <code>j</code> is the site index, <code>speccov1</code>
is a continuous species covariate, and <code>sitecov1</code> and
<code>sitecov2</code> are continuous site covariates. <code>psi</code>
can be modeled as a function of the species and site covariates.</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th align="left"><code>formula_psi</code></th>
<th align="left"><code>formula_psi_shared</code></th>
<th align="left">Linear predictor specified</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(psi[i]) = gamma[i, 1]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ sitecov1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(psi[i, j]) = gamma[i, 1] + gamma[i, 2] * sitecov1[j]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ sitecov1 + sitecov2</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(psi[i, j]) = gamma[i, 1] + gamma[i, 2] * sitecov1[j] + gamma[i, 3] * sitecov2[j]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ sitecov1 * sitecov2</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(psi[i, j]) = gamma[i, 1] + gamma[i, 2] * sitecov1[j] + gamma[i, 3] * sitecov2[j] + gamma[i, 4] * sitecov1[j] * sitecov2[j]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ speccov1</code></td>
<td align="left"><code>logit(psi[i]) = gamma[i, 1] + gamma_shared[1] * speccov1[i]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ sitecov1</code></td>
<td align="left"><code>logit(psi[i, j]) = gamma[i, 1] + gamma_shared[1] * sitecov1[j]</code></td>
</tr>
</tbody>
</table>
<p>In <code><a href="../reference/occumb.html">occumb()</a></code>, species-specific effects on
<code>psi</code> are denoted by <code>gamma</code>, and the shared
effects on <code>psi</code> are denoted by <code>gamma_shared</code>.
The first row of the table specifies a default intercept-only model
where <code>logit(psi[i])</code> is determined only by the intercept
term <code>gamma[i, 1]</code>. As in this most straightforward case,
<code><a href="../reference/occumb.html">occumb()</a></code> always estimates the species-specific intercept
<code>gamma[i, 1]</code>. In the second case, the species-specific
effect <code>gamma[i, 2]</code> of the site covariate
<code>sitecov1</code> are incorporated. Note that the site subscript
<code>j</code> is added to <code>psi</code> on the left-hand side of the
equation because the value of <code>psi</code> now varies from site to
site depending on the value of <code>sitecov1[j]</code>. In the third
and fourth cases, another site covariate, <code>sitecov2</code>, is
specified in addition to <code>sitecov1</code>. In the fourth case,
interaction is specified using the <code>*</code> operator.</p>
<p>In the fifth case, the <code>formula_psi_shared</code> argument
specifies the shared effect of the species covariate
<code>speccov1</code>. Note that the effect <code>gamma_shared[1]</code>
of <code>speccov1[i]</code> in the linear predictor does not have
subscript <code>i</code>. Because species-specific effects cannot be
estimated for species covariates, <code><a href="../reference/occumb.html">occumb()</a></code> accepts species
covariates and their interactions only in its
<code>formula_&lt;parameter name&gt;_shared</code> argument. Introducing
species covariates does not change the dimension of <code>psi</code>
(note that it has only the subscript <code>i</code>), but may help
reveal variations in site occupancy probability associated with species
characteristics.</p>
<p>In the sixth case, the site covariate <code>sitecov1</code> is
specified in the <code>formula_psi_shared</code> argument. Note that, in
contrast to the second case, <code>sitecov1[j]</code> has a shared
effect <code>gamma_shared[1]</code>. Because species are often expected
to respond differently to site characteristics, site covariates are
likely to be introduced using the <code>formula_psi</code> argument.
Nevertheless, the <code>formula_psi_shared</code> argument can be used
when consistent covariate effects across species are expected or when
the data support doing so.</p>
<p>A similar approach can be applied to <code>theta</code> and
<code>phi</code>, which can be modeled as functions of species, site,
and replicate covariates. The following table shows examples of
<code>theta</code> modeling, where <code>i</code> is the species index,
<code>j</code> is the site index, <code>k</code> is the replicate index,
<code>speccov1</code> is a continuous species covariate,
<code>sitecov1</code> is a continuous site covariate, and
<code>replcov1</code> is a continuous replicate covariate.</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th align="left"><code>formula_theta</code></th>
<th align="left"><code>formula_theta_shared</code></th>
<th align="left">Linear predictor specified</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(theta[i]) = beta[i, 1]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ sitecov1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(theta[i, j]) = beta[i, 1] + beta[i, 2] * sitecov1[j]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ replcov1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>logit(theta[i, j, k]) = beta[i, 1] + beta[i, 2] * replcov1[j, k]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ speccov1</code></td>
<td align="left"><code>logit(theta[i]) = beta[i, 1] + beta_shared[1] * speccov1[i]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ sitecov1</code></td>
<td align="left"><code>logit(theta[i, j]) = beta[i, 1] + beta_shared[1] * sitecov1[j]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ replcov1</code></td>
<td align="left"><code>logit(theta[i, j, k]) = beta[i, 1] + beta_shared[1] * replcov1[j, k]</code></td>
</tr>
</tbody>
</table>
<p>In <code><a href="../reference/occumb.html">occumb()</a></code>, species-specific effects on
<code>theta</code> are denoted as <code>beta</code> and shared effects
on <code>theta</code> are denoted as <code>beta_shared</code>. The first
row of the above table specifies an intercept-only model. As in the case
of <code>psi</code>, <code><a href="../reference/occumb.html">occumb()</a></code> always estimates the
species-specific intercept <code>beta[i, 1]</code>. The second and third
cases can be contrasted with the second case of the <code>psi</code>
example with a single covariate specified in the
<code>formula_psi</code> argument, and the remaining cases with the
fifth and sixth cases of the <code>psi</code> example with a single
covariate specified in the <code>formula_psi_shared</code> argument.
Because the replicate covariate <code>replcov1</code> has both site
index <code>j</code> and replicate index <code>k</code>, specifying it
adds these two indices to <code>theta</code>.</p>
<p>The same rule applies to the <code>phi</code> modeling. The following
is an example of a more complex case involving interactions between
different types of covariates.</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th align="left"><code>formula_phi</code></th>
<th align="left"><code>formula_phi_shared</code></th>
<th align="left">Linear predictor specified</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>~ 1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>log(phi[i]) = alpha[i, 1]</code></td>
</tr>
<tr class="even">
<td align="left"><code>~ sitecov1 * replcov1</code></td>
<td align="left"><code>~ 1</code></td>
<td align="left"><code>log(phi[i, j, k]) = alpha[i, 1] + alpha[i, 2] * sitecov1[j] + alpha[i, 3] * replcov1[j, k] + alpha[i, 4] * sitecov1[j] * replcov1[j, k]</code></td>
</tr>
<tr class="odd">
<td align="left"><code>~ replcov1</code></td>
<td align="left"><code>~ speccov1 * sitecov1</code></td>
<td align="left"><code>log(phi[i, j, k]) = alpha[i, 1] + alpha[i, 2] * replcov1[j, k] + alpha_shared[1] * speccov1[i] + alpha_shared[2] * sitecov1[j] + alpha_shared[3] * speccov1[i] * sitecov1[j]</code></td>
</tr>
</tbody>
</table>
<p>In <code><a href="../reference/occumb.html">occumb()</a></code>, species-specific effects on
<code>phi</code> are denoted as <code>alpha</code> and shared effects on
<code>phi</code> are denoted as <code>alpha_shared</code>. Similar to
the other two parameters, <code><a href="../reference/occumb.html">occumb()</a></code> always estimates the
species-specific intercept <code>alpha[i, 1]</code>.</p>
<p>The following table summarizes the covariate types accepted by each
<code>formula</code> argument.</p>
<table class="table">
<thead><tr class="header">
<th align="left">Argument</th>
<th align="center"><code>spec_cov</code></th>
<th align="center"><code>site_cov</code></th>
<th align="center"><code>repl_cov</code></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>formula_phi</code></td>
<td align="center"></td>
<td align="center">✓</td>
<td align="center">✓</td>
</tr>
<tr class="even">
<td align="left"><code>formula_theta</code></td>
<td align="center"></td>
<td align="center">✓</td>
<td align="center">✓</td>
</tr>
<tr class="odd">
<td align="left"><code>formula_psi</code></td>
<td align="center"></td>
<td align="center">✓</td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="left"><code>formula_phi_shared</code></td>
<td align="center">✓</td>
<td align="center">✓</td>
<td align="center">✓</td>
</tr>
<tr class="odd">
<td align="left"><code>formula_theta_shared</code></td>
<td align="center">✓</td>
<td align="center">✓</td>
<td align="center">✓</td>
</tr>
<tr class="even">
<td align="left"><code>formula_psi_shared</code></td>
<td align="center">✓</td>
<td align="center">✓</td>
<td align="center"></td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="prior-distributions">Prior distributions<a class="anchor" aria-label="anchor" href="#prior-distributions"></a>
</h2>
<p>A hierarchical prior distribution is specified for the
species-specific effects <code>alpha</code>, <code>beta</code>, and
<code>gamma</code>. Specifically, the vector of these effects is assumed
to follow a multivariate normal distribution, and a prior distribution
is specified for the elements of its mean vector <code>Mu</code> and
covariance matrix <code>Sigma</code>. The element values of
<code>Mu</code> and <code>Sigma</code> are estimated from the data; as
these hyperparameters summarize the variation in species-specific
effects at the community level, their estimates may be of interest in
assessing e.g., whether covariates have a consistent effect on a wide
range of species.</p>
<p>For each element of <code>Mu</code>, a normal prior distribution with
a mean of 0 and precision (i.e., the inverse of the variance)
<code>prior_prec</code> is specified. The <code>prior_prec</code> value
is determined by the <code>prior_prec</code> argument of the
<code><a href="../reference/occumb.html">occumb()</a></code> function, which by default is set to a small value
of <code>1e-4</code> to specify vague priors.</p>
<p><code>Sigma</code> is decomposed into the elements of standard
deviation <code>sigma</code> and correlation coefficient
<code>rho</code>, each of which is specified by a different vague prior.
Specifically, a uniform prior distribution with a lower limit of 0 and
an upper limit of <code>prior_ulim</code> is specified for
<code>sigma</code>, and a uniform prior with a lower limit of −1 and an
upper limit of 1 is set for <code>rho</code>. The value of
<code>prior_ulim</code> is determined by the <code>prior_ulim</code>
argument of the <code><a href="../reference/occumb.html">occumb()</a></code> function and is set to
<code>1e4</code> by default.</p>
<p>For each of the shared effects <code>alpha_shared</code>,
<code>beta_shared</code>, and <code>gamma_shared</code>, a normal prior
distribution with mean 0 and precision <code>prior_prec</code> is
specified.</p>
</div>
<div class="section level2">
<h2 id="after-all-which-are-the-relevant-parameters">After all, which are the relevant parameters?<a class="anchor" aria-label="anchor" href="#after-all-which-are-the-relevant-parameters"></a>
</h2>
<p>The latent variables and parameters of the model to be estimated and
saved using the <code><a href="../reference/occumb.html">occumb()</a></code> function are as follows. Note that
<code><a href="../reference/occumb.html">occumb()</a></code> will not save <code>u</code> and <code>r</code>,
but their function <code>pi</code>. The posterior samples of these
latent variables and parameters can be accessed using
<code><a href="../reference/get_posterior.html">get_post_samples()</a></code> or <code><a href="../reference/get_posterior.html">get_post_summary()</a></code>
functions.</p>
<dl>
<dt>
<code>z</code>
</dt>
<dd>
Site occupancy status of species.
</dd>
<dt>
<code>pi</code>
</dt>
<dd>
Multinomial probabilities of species sequence read counts.
</dd>
<dt>
<code>phi</code>
</dt>
<dd>
Sequence relative dominance of species.
</dd>
<dt>
<code>theta</code>
</dt>
<dd>
Sequence capture probabilities of species.
</dd>
<dt>
<code>psi</code>
</dt>
<dd>
Site occupancy probabilities of species.
</dd>
<dt>
<code>alpha</code>
</dt>
<dd>
Species-specific effects on sequence relative dominance
(<code>phi</code>).
</dd>
<dt>
<code>beta</code>
</dt>
<dd>
Species-specific effects on sequence capture probabilities
(<code>theta</code>).
</dd>
<dt>
<code>gamma</code>
</dt>
<dd>
Species-specific effects on site occupancy probabilities
(<code>psi</code>).
</dd>
<dt>
<code>alpha_shared</code>
</dt>
<dd>
Effects on sequence relative dominance (<code>phi</code>) common across
species.
</dd>
<dt>
<code>beta_shared</code>
</dt>
<dd>
Effects on sequence capture probabilities (<code>theta</code>) that are
common across species.
</dd>
<dt>
<code>gamma_shared</code>
</dt>
<dd>
Effects on site occupancy probabilities (<code>psi</code>) that are
common across species.
</dd>
<dt>
<code>Mu</code>
</dt>
<dd>
Community-level averages of species-specific effects
(<code>alpha</code>, <code>beta</code>, <code>gamma</code>).
</dd>
<dt>
<code>sigma</code>
</dt>
<dd>
Standard deviations of species-specific effects (<code>alpha</code>,
<code>beta</code>, <code>gamma</code>).
</dd>
<dt>
<code>rho</code>
</dt>
<dd>
Correlation coefficients of the species-specific effects
(<code>alpha</code>, <code>beta</code>, <code>gamma</code>).
</dd>
</dl>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Keiichi Fukaya.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
