<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to occumb • occumb</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to occumb">
<meta name="robots" content="noindex">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">occumb</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.1.0.9007</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/occumb.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/model_specification.html">Model specification in occumb()</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/fukayak/occumb/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to occumb</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/fukayak/occumb/blob/develop/vignettes/occumb.Rmd" class="external-link"><code>vignettes/occumb.Rmd</code></a></small>
      <div class="d-none name"><code>occumb.Rmd</code></div>
    </div>

    
    
<p>This section briefly describes the steps involved in the series of
model-based analyses of eDNA metabarcoding data using the
<code>occumb</code> R package.</p>
<div class="section level2">
<h2 id="prepare-dataset">Prepare dataset<a class="anchor" aria-label="anchor" href="#prepare-dataset"></a>
</h2>
<p>We will use the package’s built-in data <code>fish</code> (see
<code><a href="../reference/fish.html">?fish</a></code> for documentation) to analyze the sequence count data
using the <code>occumb</code> package. The <code><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary()</a></code>
function can be used to provide an overview of the dataset.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">fish</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">fish</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Sequence read counts: </span></span>
<span><span class="co">#&gt;  Number of species, I = 50 </span></span>
<span><span class="co">#&gt;  Number of sites, J = 50 </span></span>
<span><span class="co">#&gt;  Maximum number of replicates per site, K = 3 </span></span>
<span><span class="co">#&gt;  Number of missing observations = 6 </span></span>
<span><span class="co">#&gt;  Number of replicates per site: 2.88 (average), 0.33 (sd)</span></span>
<span><span class="co">#&gt;  Sequencing depth: 77910 (average), 98034.7 (sd)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Species covariates: </span></span>
<span><span class="co">#&gt;  mismatch (continuous) </span></span>
<span><span class="co">#&gt; Site covariates: </span></span>
<span><span class="co">#&gt;  riverbank (categorical) </span></span>
<span><span class="co">#&gt; Replicate covariates: </span></span>
<span><span class="co">#&gt;  (None) </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Labels for species: </span></span>
<span><span class="co">#&gt;  Abbottina rivularis, Acanthogobius lactipes, Acheilognathus macropterus, Acheilognathus rhombeus, Anguilla japonica, Biwia zezera, Carassius cuvieri, Carassius spp., Channa argus, Ctenopharyngodon idella, Cyprinus carpio, Gambusia affinis, Gnathopogon spp., Gymnogobius castaneus, Gymnogobius petschiliensis, Gymnogobius urotaenia, Hemibarbus spp., Hypomesus nipponensis, Hypophthalmichthys spp., Hyporhamphus intermedius, Ictalurus punctatus, Ischikauia steenackeri, Lepomis macrochirus macrochirus, Leucopsarion petersii, Megalobrama amblycephala, Micropterus dolomieu dolomieu, Micropterus salmoides, Misgurnus spp., Monopterus albus, Mugil cephalus cephalus, Mylopharyngodon piceus, Nipponocypris sieboldii, Nipponocypris temminckii, Opsariichthys platypus, Opsariichthys uncirostris uncirostris, Oryzias latipes, Plecoglossus altivelis altivelis, Pseudogobio spp., Pseudorasbora parva, Rhinogobius spp., Rhodeus ocellatus ocellatus, Salangichthys microdon, Sarcocheilichthys variegatus microoculus, Silurus asotus, Squalidus chankaensis biwae, Tachysurus tokiensis, Tanakia lanceolata, Tribolodon brandtii maruta, Tribolodon hakonensis, Tridentiger spp. </span></span>
<span><span class="co">#&gt; Labels for sites: </span></span>
<span><span class="co">#&gt;  1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50 </span></span>
<span><span class="co">#&gt; Labels for replicates: </span></span>
<span><span class="co">#&gt;  L, C, R </span></span></code></pre></div>
<p>A summary of the sequence read count data can be found in the first
block of the output. Data were obtained from samples taken from 50 sites
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>×</mo><annotation encoding="application/x-tex">\times</annotation></semantics></math>
(up to) three replicates, with 50 fish species recorded. A summary of
the missing samples, number of replicates per site, and sequencing depth
(i.e., the total number of sequence reads per sample) is also presented.
The second block of the output indicates that the <code>fish</code> data
also have two covariates: one continuous species covariate,
<code>mismatch</code>, and one discrete site covariate,
<code>riverbank</code>, which can be used in the following analysis. The
third block of the output shows the labels assigned to the 50 species,
50 sites, and three replicates.</p>
<p>As such, the <code>fish</code> data brought together a set of data
relevant to the analysis using the <code>occumb</code> package. When
analyzing one’s own dataset, you can use <code><a href="../reference/occumbData.html">occumbData()</a></code> to
set up a data object.</p>
<p>If you want to display the entire data, just type
<code>fish</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fish</span> <span class="co"># Result not shown</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fit-models">Fit models<a class="anchor" aria-label="anchor" href="#fit-models"></a>
</h2>
<p>Using the <code>fish</code> dataset (or the dataset constructed with
<code><a href="../reference/occumbData.html">occumbData()</a></code>), one can fit a multispecies site occupancy
model for eDNA metabarcoding using the <code><a href="../reference/occumb.html">occumb()</a></code> function
which fits the model in a fully Bayesian approach via the Markov Chain
Monte Carlo (MCMC) method implemented in <a href="https://mcmc-jags.sourceforge.io/" class="external-link">JAGS</a>.</p>
<p><code><a href="../reference/occumb.html">occumb()</a></code> has several arguments that accept model
formulas to fit different model variants using the usual R formula
syntax. See <code><a href="../articles/model_specification.html">vignette("model_specification")</a></code> for an overview
of the models that <code><a href="../reference/occumb.html">occumb()</a></code> can fit and details on how to
specify models.</p>
<p>Fitting a null model (i.e., an intercept-only model, although the
value of the intercept is species-specific) is very simple, as
follows:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit0</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/occumb.html">occumb</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">fish</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>This model fitting can take approximately one hour. Given that
<code><a href="../reference/occumb.html">occumb()</a></code> fits a class of complex hierarchical models with
many parameters and unknown variables, it may require, depending on the
dataset size, a long time and/or a lot of memory for model fitting.
Setting <code>parallel = TRUE</code> is recommended for faster model
fitting through parallel computations.</p>
<p>A model incorporating species and site covariates from the
<code>fish</code> dataset can be fitted as follows:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/occumb.html">occumb</a></span><span class="op">(</span>formula_psi <span class="op">=</span> <span class="op">~</span> <span class="va">riverbank</span>,</span>
<span>               formula_phi_shared <span class="op">=</span> <span class="op">~</span> <span class="va">mismatch</span>,</span>
<span>               data <span class="op">=</span> <span class="va">fish</span>,</span>
<span>               parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Here, the <code>riverbank</code> is specified as the covariate for
site occupancy probability (<code>psi</code>) in the
<code>formula_psi</code> argument, and <code>mismatch</code> as the
covariate for the relative dominance of sequence (<code>phi</code>) in
the <code>formula_phi_shared</code> argument. In the latter, since
<code>mismatch</code> is a species covariate, the
<code>formula_phi_shared</code> argument is used instead of
<code>formula_phi</code> so that the effect of <code>mismatch</code> is
constant across species; see
<code><a href="../articles/model_specification.html">vignette("model_specification")</a></code> for more details.</p>
<p>Thus, as in the generalized linear models (GLMs), model parameters
can be expressed as functions of covariates, and <code><a href="../reference/occumb.html">occumb()</a></code>
allows straightforward covariate specification using the formula syntax.
We will see later how to examine the estimated parameter values and
covariate effects.</p>
<p><code><a href="../reference/occumb.html">occumb()</a></code> also has several arguments for controlling the
MCMC computation to fit the model. For example, to obtain more precise
posterior estimates than in the above example, one can explicitly set
<code>n.iter</code> and <code>n.thin</code> arguments to obtain longer
and less autocorrelated MCMC samples. In the following,
<code>n.iter</code> and <code>n.thin</code> are set to twice the
default.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit1x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/occumb.html">occumb</a></span><span class="op">(</span>formula_psi <span class="op">=</span> <span class="op">~</span> <span class="va">riverbank</span>,</span>
<span>                formula_phi_shared <span class="op">=</span> <span class="op">~</span> <span class="va">mismatch</span>,</span>
<span>                data <span class="op">=</span> <span class="va">fish</span>,</span>
<span>                n.thin <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                n.iter <span class="op">=</span> <span class="fl">40000</span>,</span>
<span>                parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Note that when fitting a model using <code><a href="../reference/occumb.html">occumb()</a></code>, a warning
message <code>"At least one Rhat value could not be calculated."</code>
may appear. Usually, this warning can be ignored safely. In models fit
by <code><a href="../reference/occumb.html">occumb()</a></code>, the values of some latent variables may not
change across MCMC iterations, making it impossible to compute
<code>Rhat</code> values for them. This is the correct behavior of the
model, but it can cause the above warning to be displayed even when the
MCMC method is working properly.</p>
</div>
<div class="section level2">
<h2 id="check-posterior-samples">Check posterior samples<a class="anchor" aria-label="anchor" href="#check-posterior-samples"></a>
</h2>
<p>With the <code>fit1</code> object, we will examine how to access and
diagnose the posterior sample obtained using <code><a href="../reference/occumb.html">occumb()</a></code>.</p>
<p>Enter the object name <code>fit1</code> to display a summary of the
MCMC results, including a table of posterior estimates for each saved
parameter.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit1</span> <span class="co"># Result not shown</span></span></code></pre></div>
<p>The <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> function draws a trace plot and density plot
for each parameter.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit1</span><span class="op">)</span> <span class="co"># Outputs many figures; enter "Esc" or "Ctrl + C" to quit</span></span></code></pre></div>
<p><img src="images%2Fplot_fit1.png"></p>
<p>Users of the <a href="https://kenkellner.com/jagsUI/" class="external-link">jagsUI</a> package should be familiar with
the outputs of these operations. Model fitting in <code><a href="../reference/occumb.html">occumb()</a></code>
relies on functions provided by the <a href="https://kenkellner.com/jagsUI/" class="external-link">jagsUI</a> package, and
the above operations apply <code>print</code> and <code>plot</code>
methods to <a href="https://kenkellner.com/jagsUI/" class="external-link">jagsUI</a> objects.</p>
<p>The <code><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary()</a></code> function provides a high-level summary of
the results. This is based on a previous version of the
<code>summary</code> method for <a href="https://kenkellner.com/jagsUI/" class="external-link">jagsUI</a> objects
(implemented until <a href="https://kenkellner.com/jagsUI/" class="external-link">jagsUI</a> package version 1.5.2) and is
modified for the <code><a href="../reference/occumb.html">occumb()</a></code> output, which displays a data
summary, model equations, MCMC settings, and a quick overview of the
convergence of the relevant model parameters.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">fit1</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Summary for an occumbFit object </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Summary of data:</span></span>
<span><span class="co">#&gt;  Number of species, I = 50 </span></span>
<span><span class="co">#&gt;  Number of sites, J = 50 </span></span>
<span><span class="co">#&gt;  Maximum number of replicates per site, K = 3 </span></span>
<span><span class="co">#&gt;  Number of missing observations = 6 </span></span>
<span><span class="co">#&gt;  Number of replicates per site: 2.88 (average), 0.33 (sd) </span></span>
<span><span class="co">#&gt;  Sequencing depth: 77910 (average), 98034.7 (sd) </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Model specification:</span></span>
<span><span class="co">#&gt;  formula_phi:          ~ 1 </span></span>
<span><span class="co">#&gt;  formula_theta:        ~ 1 </span></span>
<span><span class="co">#&gt;  formula_psi:          ~ riverbank </span></span>
<span><span class="co">#&gt;  formula_phi_shared:   ~ mismatch </span></span>
<span><span class="co">#&gt;  formula_theta_shared: ~ 1 </span></span>
<span><span class="co">#&gt;  formula_psi_shared:   ~ 1 </span></span>
<span><span class="co">#&gt;  prior_prec:           1e-04 </span></span>
<span><span class="co">#&gt;  prior_ulim:           10000 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Saved parameters:</span></span>
<span><span class="co">#&gt;  Mu sigma rho alpha beta gamma alpha_shared phi theta psi z pi deviance </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MCMC ran in parallel for 43.593 minutes at time 2023-09-11 20:16:31.787026:</span></span>
<span><span class="co">#&gt;  For each of 4 chains:</span></span>
<span><span class="co">#&gt;   Adaptation:            100 iterations (sufficient)</span></span>
<span><span class="co">#&gt;   Burn-in:               10000 iterations</span></span>
<span><span class="co">#&gt;   Thin rate:             10 iterations</span></span>
<span><span class="co">#&gt;   Total chain length:    20100 iterations</span></span>
<span><span class="co">#&gt;   Posterior sample size: 1000 draws</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Summary of posterior samples: </span></span>
<span><span class="co">#&gt;  Mu: </span></span>
<span><span class="co">#&gt;   Number of parameters: 4 </span></span>
<span><span class="co">#&gt;   Rhat:  1.009 (min), 1.027 (median), 1.043 (mean), 1.109 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 30 (min), 117.5 (median), 154.2 (mean), 352 (max) </span></span>
<span><span class="co">#&gt;  sigma: </span></span>
<span><span class="co">#&gt;   Number of parameters: 4 </span></span>
<span><span class="co">#&gt;   Rhat:  1.004 (min), 1.04 (median), 1.074 (mean), 1.213 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 17 (min), 67.5 (median), 217 (mean), 716 (max) </span></span>
<span><span class="co">#&gt;  rho: </span></span>
<span><span class="co">#&gt;   Number of parameters: 6 </span></span>
<span><span class="co">#&gt;   Rhat:  1.042 (min), 1.107 (median), 1.106 (mean), 1.187 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 19 (min), 33.5 (median), 40.7 (mean), 79 (max) </span></span>
<span><span class="co">#&gt;  alpha: </span></span>
<span><span class="co">#&gt;   Number of parameters: 50 </span></span>
<span><span class="co">#&gt;   Rhat:  1.027 (min), 1.093 (median), 1.157 (mean), 1.77 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 7 (min), 34 (median), 40.6 (mean), 104 (max) </span></span>
<span><span class="co">#&gt;  beta: </span></span>
<span><span class="co">#&gt;   Number of parameters: 50 </span></span>
<span><span class="co">#&gt;   Rhat:  1.002 (min), 1.018 (median), 1.028 (mean), 1.237 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 16 (min), 183.5 (median), 349.9 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  gamma: </span></span>
<span><span class="co">#&gt;   Number of parameters: 100 </span></span>
<span><span class="co">#&gt;   Rhat:  1.001 (min), 1.019 (median), 1.029 (mean), 1.136 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 24 (min), 173.5 (median), 363.5 (mean), 2756 (max) </span></span>
<span><span class="co">#&gt;  alpha_shared: </span></span>
<span><span class="co">#&gt;   Number of parameters: 1 </span></span>
<span><span class="co">#&gt;   Rhat:  1.1 </span></span>
<span><span class="co">#&gt;   n.eff: 31 </span></span>
<span><span class="co">#&gt;  phi: </span></span>
<span><span class="co">#&gt;   Number of parameters: 50 </span></span>
<span><span class="co">#&gt;   Rhat:  1.001 (min), 1.07 (median), 1.127 (mean), 1.664 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 8 (min), 42.5 (median), 202.7 (mean), 3161 (max) </span></span>
<span><span class="co">#&gt;  theta: </span></span>
<span><span class="co">#&gt;   Number of parameters: 50 </span></span>
<span><span class="co">#&gt;   Rhat:  1.002 (min), 1.016 (median), 1.024 (mean), 1.219 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 16 (min), 196 (median), 358.1 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  psi: </span></span>
<span><span class="co">#&gt;   Number of parameters: 2500 </span></span>
<span><span class="co">#&gt;   Rhat:  1 (min), 1.008 (median), 1.013 (mean), 1.101 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 36 (min), 396 (median), 679.3 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  z: </span></span>
<span><span class="co">#&gt;   Number of parameters: 2500 </span></span>
<span><span class="co">#&gt;   Rhat:  1 (min), 1.006 (median), 1.009 (mean), 1.184 (max), 946 (Number of NAs) </span></span>
<span><span class="co">#&gt;   n.eff: 1 (min), 1233.5 (median), 1628.6 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  pi: </span></span>
<span><span class="co">#&gt;   Number of parameters: 7500 </span></span>
<span><span class="co">#&gt;   Rhat:  1 (min), 1.078 (median), 1.102 (mean), 1.294 (max), 30 (Number of NAs) </span></span>
<span><span class="co">#&gt;   n.eff: 1 (min), 1 (median), 1009.1 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  deviance: </span></span>
<span><span class="co">#&gt;   Rhat:  1.001 </span></span>
<span><span class="co">#&gt;   n.eff: 1332 </span></span></code></pre></div>
<p>Convergence of the MCMC algorithm is determined based on the
<code>Rhat</code> value, which becomes 1 when the chains are converged.
Customarily, <code>Rhat</code> values of less than 1.1 will be required
for the parameters of interest. Overall, the <code>fit1</code> model is
not very far from convergent, but the convergence of <code>phi</code>
and <code>alpha</code>, the parameters related to the relative dominance
of species sequence reads, is not good (i.e., high values of
<code>Rhat</code>); hence, longer MCMC runs will be required for a
formal inference.</p>
<p>The <code><a href="../reference/get_posterior.html">get_post_samples()</a></code> and
<code><a href="../reference/get_posterior.html">get_post_summary()</a></code> functions can be used to access the
posterior samples of individual parameters or their summaries. This may
require understanding the model parameters and their names: see
<code><a href="../articles/model_specification.html">vignette("model_specification")</a></code> for details. As an example,
we attempt to obtain a posterior summary of the site occupancy
probability <code>psi</code> using the <code><a href="../reference/get_posterior.html">get_post_summary()</a></code>
function.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_summary_psi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_posterior.html">get_post_summary</a></span><span class="op">(</span><span class="va">fit1</span>, <span class="st">"psi"</span><span class="op">)</span></span>
<span><span class="va">post_summary_psi</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;                  mean          sd        2.5%        25%        50%        75%      97.5%     Rhat n.eff overlap0 f</span></span>
<span><span class="co">#&gt; psi[1,1]   0.64607375 0.083034133 0.466158634 0.59382744 0.65078275 0.70478301 0.79318634 1.001697  1264        0 1</span></span>
<span><span class="co">#&gt; psi[2,1]   0.28250539 0.089275014 0.136970706 0.21789079 0.27480943 0.33672122 0.47971573 1.013796   317        0 1</span></span>
<span><span class="co">#&gt; psi[3,1]   0.38321762 0.088583160 0.218908875 0.32150169 0.38055944 0.44171514 0.56510758 1.005127   472        0 1</span></span>
<span><span class="co">#&gt; psi[4,1]   0.16899696 0.070833671 0.058363836 0.11684541 0.15943020 0.21075828 0.32821669 1.001143  3788        0 1</span></span>
<span><span class="co">#&gt; psi[5,1]   0.20435372 0.088327893 0.069729103 0.13950376 0.19315615 0.25286705 0.41681236 1.006308   596        0 1</span></span>
<span><span class="co">#&gt; psi[6,1]   0.10737782 0.061179251 0.024299550 0.06246265 0.09573034 0.13925811 0.25577838 1.002291  1418        0 1</span></span>
<span><span class="co">#&gt; psi[7,1]   0.84639119 0.053478041 0.730770215 0.81327035 0.85048238 0.88494478 0.93751979 1.003575   654        0 1</span></span>
<span><span class="co">#&gt; psi[8,1]   0.94296846 0.028288919 0.875373936 0.92739641 0.94770976 0.96354354 0.98337253 1.002581  1006        0 1</span></span>
<span><span class="co">#&gt; (Omitted the remaining)</span></span></code></pre></div>
<p>In the <code>fit1</code> model, <code>psi</code> was assumed to have
different values for each site depending on the <code>riverbank</code>
covariate (remember the <code>psi_formula</code> argument for
<code>fit1</code>). Therefore, <code>psi</code> was estimated for each
of the 50 species
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>×</mo><annotation encoding="application/x-tex">\times</annotation></semantics></math>
50 sites (note that <code>psi</code> differs for each species by
default), and the two subscripts in brackets distinguish them. Which
species and site parameter was <code>psi[1,1]</code> about? How about
<code>psi[2,1]</code>? Information regarding the dimensions of the
parameters is provided in the attributes of the resulting object.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_summary_psi</span><span class="op">)</span><span class="op">$</span><span class="va">dimension</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1] "Species" "Site"</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_summary_psi</span><span class="op">)</span><span class="op">$</span><span class="va">label</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; $Species</span></span>
<span><span class="co">#&gt;  [1] "Abbottina rivularis"                     </span></span>
<span><span class="co">#&gt;  [2] "Acanthogobius lactipes"                  </span></span>
<span><span class="co">#&gt;  [3] "Acheilognathus macropterus"              </span></span>
<span><span class="co">#&gt;  [4] "Acheilognathus rhombeus"                 </span></span>
<span><span class="co">#&gt;  [5] "Anguilla japonica"                       </span></span>
<span><span class="co">#&gt;  [6] "Biwia zezera"                            </span></span>
<span><span class="co">#&gt;  [7] "Carassius cuvieri"                       </span></span>
<span><span class="co">#&gt;  [8] "Carassius spp."                          </span></span>
<span><span class="co">#&gt;  [9] "Channa argus"                            </span></span>
<span><span class="co">#&gt; [10] "Ctenopharyngodon idella"                 </span></span>
<span><span class="co">#&gt; [11] "Cyprinus carpio"                         </span></span>
<span><span class="co">#&gt; [12] "Gambusia affinis"                        </span></span>
<span><span class="co">#&gt; [13] "Gnathopogon spp."                        </span></span>
<span><span class="co">#&gt; [14] "Gymnogobius castaneus"                   </span></span>
<span><span class="co">#&gt; [15] "Gymnogobius petschiliensis"              </span></span>
<span><span class="co">#&gt; [16] "Gymnogobius urotaenia"                   </span></span>
<span><span class="co">#&gt; [17] "Hemibarbus spp."                         </span></span>
<span><span class="co">#&gt; [18] "Hypomesus nipponensis"                   </span></span>
<span><span class="co">#&gt; [19] "Hypophthalmichthys spp."                 </span></span>
<span><span class="co">#&gt; [20] "Hyporhamphus intermedius"                </span></span>
<span><span class="co">#&gt; [21] "Ictalurus punctatus"                     </span></span>
<span><span class="co">#&gt; [22] "Ischikauia steenackeri"                  </span></span>
<span><span class="co">#&gt; [23] "Lepomis macrochirus macrochirus"         </span></span>
<span><span class="co">#&gt; [24] "Leucopsarion petersii"                   </span></span>
<span><span class="co">#&gt; [25] "Megalobrama amblycephala"                </span></span>
<span><span class="co">#&gt; [26] "Micropterus dolomieu dolomieu"           </span></span>
<span><span class="co">#&gt; [27] "Micropterus salmoides"                   </span></span>
<span><span class="co">#&gt; [28] "Misgurnus spp."                          </span></span>
<span><span class="co">#&gt; [29] "Monopterus albus"                        </span></span>
<span><span class="co">#&gt; [30] "Mugil cephalus cephalus"                 </span></span>
<span><span class="co">#&gt; [31] "Mylopharyngodon piceus"                  </span></span>
<span><span class="co">#&gt; [32] "Nipponocypris sieboldii"                 </span></span>
<span><span class="co">#&gt; [33] "Nipponocypris temminckii"                </span></span>
<span><span class="co">#&gt; [34] "Opsariichthys platypus"                  </span></span>
<span><span class="co">#&gt; [35] "Opsariichthys uncirostris uncirostris"   </span></span>
<span><span class="co">#&gt; [36] "Oryzias latipes"                         </span></span>
<span><span class="co">#&gt; [37] "Plecoglossus altivelis altivelis"        </span></span>
<span><span class="co">#&gt; [38] "Pseudogobio spp."                        </span></span>
<span><span class="co">#&gt; [39] "Pseudorasbora parva"                     </span></span>
<span><span class="co">#&gt; [40] "Rhinogobius spp."                        </span></span>
<span><span class="co">#&gt; [41] "Rhodeus ocellatus ocellatus"             </span></span>
<span><span class="co">#&gt; [42] "Salangichthys microdon"                  </span></span>
<span><span class="co">#&gt; [43] "Sarcocheilichthys variegatus microoculus"</span></span>
<span><span class="co">#&gt; [44] "Silurus asotus"                          </span></span>
<span><span class="co">#&gt; [45] "Squalidus chankaensis biwae"             </span></span>
<span><span class="co">#&gt; [46] "Tachysurus tokiensis"                    </span></span>
<span><span class="co">#&gt; [47] "Tanakia lanceolata"                      </span></span>
<span><span class="co">#&gt; [48] "Tribolodon brandtii maruta"              </span></span>
<span><span class="co">#&gt; [49] "Tribolodon hakonensis"                   </span></span>
<span><span class="co">#&gt; [50] "Tridentiger spp."                        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Site</span></span>
<span><span class="co">#&gt;  [1] "1"  "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9"  "10" "11" "12" "13" "14" "15"</span></span>
<span><span class="co">#&gt; [16] "16" "17" "18" "19" "20" "21" "22" "23" "24" "25" "26" "27" "28" "29" "30"</span></span>
<span><span class="co">#&gt; [31] "31" "32" "33" "34" "35" "36" "37" "38" "39" "40" "41" "42" "43" "44" "45"</span></span>
<span><span class="co">#&gt; [46] "46" "47" "48" "49" "50"</span></span></code></pre></div>
<p>Thus, from the <code>$dimension</code> attribute, one can know that
the first subscript of <code>psi</code> represents species and the
second sites. From the <code>$label</code> attribute, one can know that
<code>psi[1,1]</code> represents the value of <code>psi</code> for
<em>Abbottina rivularis</em> at site 1, and <code>psi[2,1]</code>
represents the value of <code>psi</code> for <em>Acanthogobius
lactipes</em> at site 1.</p>
<p>Let us also look at the posterior summary of <code>gamma</code>,
which denotes species-specific effects on <code>psi</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_summary_gamma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_posterior.html">get_post_summary</a></span><span class="op">(</span><span class="va">fit1</span>, <span class="st">"gamma"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_summary_gamma</span><span class="op">)</span><span class="op">$</span><span class="va">dimension</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1] "Species" "Effects"</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_summary_gamma</span><span class="op">)</span><span class="op">$</span><span class="va">label</span><span class="op">$</span><span class="va">Effects</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1] "(Intercept)"                 "riverbankwithout_vegetation"</span></span></code></pre></div>
<p><code>gamma</code> had two effect values for each species: the
intercept, <code>(Intercept)</code>, and the effect of the absence of
vegetation on the riverbank, <code>riverbankwithout_vegetation</code>.
Displaying the summary, one can see that, for many species, the absence
of vegetation has a negative effect on <code>psi</code> (on its link
scale).</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_summary_gamma</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;                    mean        sd       2.5%          25%         50%         75%        97.5%      Rhat n.eff overlap0       f</span></span>
<span><span class="co">#&gt; (Omitted the beginning)</span></span>
<span><span class="co">#&gt; gamma[1,2]  -0.68135951 0.4102181 -1.4106885 -0.964385346 -0.71198673 -0.41908665  0.197784615 1.0031831   720        1 0.93875</span></span>
<span><span class="co">#&gt; gamma[2,2]  -0.57124068 0.4861383 -1.4368696 -0.906331725 -0.60645174 -0.27472631  0.480190462 1.0141750   414        1 0.87925</span></span>
<span><span class="co">#&gt; gamma[3,2]  -1.07479024 0.4260032 -1.9550037 -1.336688784 -1.07308990 -0.80888005 -0.213626578 1.0040679   632        0 0.99250</span></span>
<span><span class="co">#&gt; gamma[4,2]  -0.96563859 0.5007208 -1.9760743 -1.276917710 -0.95208744 -0.64960250  0.016721077 1.0020286  1835        1 0.97225</span></span>
<span><span class="co">#&gt; gamma[5,2]  -1.01038376 0.5436033 -2.1340409 -1.339282006 -0.98783592 -0.65193007  0.006887964 1.0148153   190        1 0.97400</span></span>
<span><span class="co">#&gt; gamma[6,2]  -0.96443244 0.5694898 -2.1291123 -1.307706597 -0.96394933 -0.62099851  0.205483803 1.0043719   989        1 0.95325</span></span>
<span><span class="co">#&gt; gamma[7,2]  -1.44522243 0.4356167 -2.4011743 -1.721131931 -1.39965212 -1.14153735 -0.689367497 1.0021543  1065        0 0.99925</span></span>
<span><span class="co">#&gt; gamma[8,2]  -1.36146823 0.4820477 -2.3661876 -1.654265660 -1.32819584 -1.04709249 -0.483307820 1.0014346  1791        0 0.99900</span></span>
<span><span class="co">#&gt; (Omitted the remaining)</span></span></code></pre></div>
<p>A posterior summary of <code>Mu</code>, which denotes the community
average of species-specific effects, shows that
<code>riverbankwithout_vegetation</code> has a community-wide negative
effect on <code>psi</code> (note from the <code>label</code> attribute
that <code>Mu[4]</code> is the community-level effect of
<code>riverbankwithout_vegetation</code>).</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_posterior.html">get_post_summary</a></span><span class="op">(</span><span class="va">fit1</span>, <span class="st">"Mu"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;              mean        sd       2.5%        25%         50%        75%</span></span>
<span><span class="co">#&gt; Mu[1] -1.18914419 0.1466486 -1.4845230 -1.2872349 -1.18205680 -1.0888031</span></span>
<span><span class="co">#&gt; Mu[2]  0.94327206 0.1945952  0.5640255  0.8102745  0.94803155  1.0691758</span></span>
<span><span class="co">#&gt; Mu[3]  0.02522906 0.3264013 -0.6034793 -0.1859071  0.01929096  0.2317505</span></span>
<span><span class="co">#&gt; Mu[4] -1.03746412 0.1609537 -1.3626390 -1.1414288 -1.03743346 -0.9292544</span></span>
<span><span class="co">#&gt;            97.5%     Rhat n.eff overlap0      f</span></span>
<span><span class="co">#&gt; Mu[1] -0.9119219 1.108793    30        0 1.0000</span></span>
<span><span class="co">#&gt; Mu[2]  1.3284452 1.031154   100        0 1.0000</span></span>
<span><span class="co">#&gt; Mu[3]  0.6951928 1.008576   352        1 0.5255</span></span>
<span><span class="co">#&gt; Mu[4] -0.7312586 1.022904   135        0 1.0000</span></span>
<span><span class="co">#&gt; attr(,"dimension")</span></span>
<span><span class="co">#&gt; [1] "Effects"</span></span>
<span><span class="co">#&gt; attr(,"label")</span></span>
<span><span class="co">#&gt; attr(,"label")$Effects</span></span>
<span><span class="co">#&gt; [1] "phi | (Intercept)"                 "theta | (Intercept)"              </span></span>
<span><span class="co">#&gt; [3] "psi | (Intercept)"                 "psi | riverbankwithout_vegetation"</span></span></code></pre></div>
<p>Next, let us access the posterior samples rather than their summary
using the <code><a href="../reference/get_posterior.html">get_post_samples()</a></code> function. We extract the
posterior sample of <code>psi</code> from <code>fit1</code> as an
example.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_sample_psi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_posterior.html">get_post_samples</a></span><span class="op">(</span><span class="va">fit1</span>, <span class="st">"psi"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">post_sample_psi</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1] 4000   50   50</span></span></code></pre></div>
<p>The resulting object, <code>post_sample_psi</code>, is a
3-dimensional array containing the posterior sample of <code>psi</code>.
Information on the dimensions of the extracted sample is provided in the
attributes.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_sample_psi</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; $dim</span></span>
<span><span class="co">#&gt; [1] 4000   50   50</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $dimension</span></span>
<span><span class="co">#&gt; [1] "Sample"  "Species" "Site"   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $label</span></span>
<span><span class="co">#&gt; $label$Sample</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $label$Species</span></span>
<span><span class="co">#&gt;  [1] "Abbottina rivularis"                     </span></span>
<span><span class="co">#&gt;  [2] "Acanthogobius lactipes"                  </span></span>
<span><span class="co">#&gt;  [3] "Acheilognathus macropterus"              </span></span>
<span><span class="co">#&gt;  [4] "Acheilognathus rhombeus"                 </span></span>
<span><span class="co">#&gt;  [5] "Anguilla japonica"                       </span></span>
<span><span class="co">#&gt;  [6] "Biwia zezera"                            </span></span>
<span><span class="co">#&gt;  [7] "Carassius cuvieri"                       </span></span>
<span><span class="co">#&gt;  [8] "Carassius spp."                          </span></span>
<span><span class="co">#&gt;  [9] "Channa argus"                            </span></span>
<span><span class="co">#&gt; [10] "Ctenopharyngodon idella"                 </span></span>
<span><span class="co">#&gt; [11] "Cyprinus carpio"                         </span></span>
<span><span class="co">#&gt; [12] "Gambusia affinis"                        </span></span>
<span><span class="co">#&gt; [13] "Gnathopogon spp."                        </span></span>
<span><span class="co">#&gt; [14] "Gymnogobius castaneus"                   </span></span>
<span><span class="co">#&gt; [15] "Gymnogobius petschiliensis"              </span></span>
<span><span class="co">#&gt; [16] "Gymnogobius urotaenia"                   </span></span>
<span><span class="co">#&gt; [17] "Hemibarbus spp."                         </span></span>
<span><span class="co">#&gt; [18] "Hypomesus nipponensis"                   </span></span>
<span><span class="co">#&gt; [19] "Hypophthalmichthys spp."                 </span></span>
<span><span class="co">#&gt; [20] "Hyporhamphus intermedius"                </span></span>
<span><span class="co">#&gt; [21] "Ictalurus punctatus"                     </span></span>
<span><span class="co">#&gt; [22] "Ischikauia steenackeri"                  </span></span>
<span><span class="co">#&gt; [23] "Lepomis macrochirus macrochirus"         </span></span>
<span><span class="co">#&gt; [24] "Leucopsarion petersii"                   </span></span>
<span><span class="co">#&gt; [25] "Megalobrama amblycephala"                </span></span>
<span><span class="co">#&gt; [26] "Micropterus dolomieu dolomieu"           </span></span>
<span><span class="co">#&gt; [27] "Micropterus salmoides"                   </span></span>
<span><span class="co">#&gt; [28] "Misgurnus spp."                          </span></span>
<span><span class="co">#&gt; [29] "Monopterus albus"                        </span></span>
<span><span class="co">#&gt; [30] "Mugil cephalus cephalus"                 </span></span>
<span><span class="co">#&gt; [31] "Mylopharyngodon piceus"                  </span></span>
<span><span class="co">#&gt; [32] "Nipponocypris sieboldii"                 </span></span>
<span><span class="co">#&gt; [33] "Nipponocypris temminckii"                </span></span>
<span><span class="co">#&gt; [34] "Opsariichthys platypus"                  </span></span>
<span><span class="co">#&gt; [35] "Opsariichthys uncirostris uncirostris"   </span></span>
<span><span class="co">#&gt; [36] "Oryzias latipes"                         </span></span>
<span><span class="co">#&gt; [37] "Plecoglossus altivelis altivelis"        </span></span>
<span><span class="co">#&gt; [38] "Pseudogobio spp."                        </span></span>
<span><span class="co">#&gt; [39] "Pseudorasbora parva"                     </span></span>
<span><span class="co">#&gt; [40] "Rhinogobius spp."                        </span></span>
<span><span class="co">#&gt; [41] "Rhodeus ocellatus ocellatus"             </span></span>
<span><span class="co">#&gt; [42] "Salangichthys microdon"                  </span></span>
<span><span class="co">#&gt; [43] "Sarcocheilichthys variegatus microoculus"</span></span>
<span><span class="co">#&gt; [44] "Silurus asotus"                          </span></span>
<span><span class="co">#&gt; [45] "Squalidus chankaensis biwae"             </span></span>
<span><span class="co">#&gt; [46] "Tachysurus tokiensis"                    </span></span>
<span><span class="co">#&gt; [47] "Tanakia lanceolata"                      </span></span>
<span><span class="co">#&gt; [48] "Tribolodon brandtii maruta"              </span></span>
<span><span class="co">#&gt; [49] "Tribolodon hakonensis"                   </span></span>
<span><span class="co">#&gt; [50] "Tridentiger spp."                        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $label$Site</span></span>
<span><span class="co">#&gt;  [1] "1"  "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9"  "10" "11" "12" "13" "14" "15"</span></span>
<span><span class="co">#&gt; [16] "16" "17" "18" "19" "20" "21" "22" "23" "24" "25" "26" "27" "28" "29" "30"</span></span>
<span><span class="co">#&gt; [31] "31" "32" "33" "34" "35" "36" "37" "38" "39" "40" "41" "42" "43" "44" "45"</span></span>
<span><span class="co">#&gt; [46] "46" "47" "48" "49" "50"</span></span></code></pre></div>
<p>Thus, the second dimension of <code>post_sample_psi</code>
corresponds to species and the third to sites. The labels for species
and sites indicate that <code>post_sample_psi[, 1, 1]</code> contains
the posterior sample of <code>psi</code> of <em>Abbottina rivularis</em>
at site 1.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_sample_psi</span><span class="op">[</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;  [1] 0.6673664 0.7505735 0.6416083 0.7011978 0.5935338 0.6087940 0.6159829</span></span>
<span><span class="co">#&gt;  [8] 0.7758206 0.6710466 0.7423708 0.6244308 0.6237291 0.6344345 0.6715390</span></span>
<span><span class="co">#&gt; [15] 0.6255393 0.7306071 0.6781538 0.7391794 0.6678208 0.6715868 0.7582302</span></span>
<span><span class="co">#&gt; [22] 0.5947020 0.4880676 0.6958859 0.6202520 0.5487584 0.6781225 0.7073072</span></span>
<span><span class="co">#&gt; (Omitted the remaining)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="predict-parameter-values">Predict parameter values<a class="anchor" aria-label="anchor" href="#predict-parameter-values"></a>
</h2>
<p>In <code><a href="../reference/occumb.html">occumb()</a></code>, three parameters, <code>psi</code>,
<code>theta</code>, and <code>phi</code>, can be modeled as a function
of covariates (see <code><a href="../articles/model_specification.html">vignette("model_specification")</a></code> for
details). Prediction of these parameters conditional on a given set of
covariate values can be easily obtained using the <code><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict()</a></code>
function to the fitted object.</p>
<p>The <code><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict()</a></code> function has the <code>newdata</code>
argument that accepts an optional dataset object used for prediction. If
<code>newdata</code> is not specified, predictions are made using the
fitted covariates; this could be an alternative to the
<code><a href="../reference/get_posterior.html">get_post_summary()</a></code> and <code><a href="../reference/get_posterior.html">get_post_samples()</a></code>
functions for accessing posterior summaries and posterior samples of
parameters. For example, the following will output a 3-dimensional array
of the posterior median and upper and lower limits of the 95% credible
interval for <code>psi</code> of the 50 species and 50 sites.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">predict_psi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">fit1</span>, parameter <span class="op">=</span> <span class="st">"psi"</span>, type <span class="op">=</span> <span class="st">"quantiles"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">predict_psi</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1]  3 50 50</span></span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">predict_psi</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; $dim</span></span>
<span><span class="co">#&gt; [1]  3 50 50</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $dimnames</span></span>
<span><span class="co">#&gt; $dimnames[[1]]</span></span>
<span><span class="co">#&gt; [1] "50%"   "2.5%"  "97.5%"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $dimnames[[2]]</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $dimnames[[3]]</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $parameter</span></span>
<span><span class="co">#&gt; [1] "psi"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $scale</span></span>
<span><span class="co">#&gt; [1] "response"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $dimension</span></span>
<span><span class="co">#&gt; [1] "Statistics" "Species"    "Sites"     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $label</span></span>
<span><span class="co">#&gt; $label$Statistics</span></span>
<span><span class="co">#&gt; [1] "50%"   "2.5%"  "97.5%"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $label$Species</span></span>
<span><span class="co">#&gt;  [1] "Abbottina rivularis"                     </span></span>
<span><span class="co">#&gt;  [2] "Acanthogobius lactipes"                  </span></span>
<span><span class="co">#&gt;  [3] "Acheilognathus macropterus"              </span></span>
<span><span class="co">#&gt;  [4] "Acheilognathus rhombeus"                 </span></span>
<span><span class="co">#&gt;  [5] "Anguilla japonica"                       </span></span>
<span><span class="co">#&gt;  [6] "Biwia zezera"                            </span></span>
<span><span class="co">#&gt;  [7] "Carassius cuvieri"                       </span></span>
<span><span class="co">#&gt;  [8] "Carassius spp."                          </span></span>
<span><span class="co">#&gt;  [9] "Channa argus"                            </span></span>
<span><span class="co">#&gt; [10] "Ctenopharyngodon idella"                 </span></span>
<span><span class="co">#&gt; [11] "Cyprinus carpio"                         </span></span>
<span><span class="co">#&gt; [12] "Gambusia affinis"                        </span></span>
<span><span class="co">#&gt; [13] "Gnathopogon spp."                        </span></span>
<span><span class="co">#&gt; [14] "Gymnogobius castaneus"                   </span></span>
<span><span class="co">#&gt; [15] "Gymnogobius petschiliensis"              </span></span>
<span><span class="co">#&gt; [16] "Gymnogobius urotaenia"                   </span></span>
<span><span class="co">#&gt; [17] "Hemibarbus spp."                         </span></span>
<span><span class="co">#&gt; [18] "Hypomesus nipponensis"                   </span></span>
<span><span class="co">#&gt; [19] "Hypophthalmichthys spp."                 </span></span>
<span><span class="co">#&gt; [20] "Hyporhamphus intermedius"                </span></span>
<span><span class="co">#&gt; [21] "Ictalurus punctatus"                     </span></span>
<span><span class="co">#&gt; [22] "Ischikauia steenackeri"                  </span></span>
<span><span class="co">#&gt; [23] "Lepomis macrochirus macrochirus"         </span></span>
<span><span class="co">#&gt; [24] "Leucopsarion petersii"                   </span></span>
<span><span class="co">#&gt; [25] "Megalobrama amblycephala"                </span></span>
<span><span class="co">#&gt; [26] "Micropterus dolomieu dolomieu"           </span></span>
<span><span class="co">#&gt; [27] "Micropterus salmoides"                   </span></span>
<span><span class="co">#&gt; [28] "Misgurnus spp."                          </span></span>
<span><span class="co">#&gt; [29] "Monopterus albus"                        </span></span>
<span><span class="co">#&gt; [30] "Mugil cephalus cephalus"                 </span></span>
<span><span class="co">#&gt; [31] "Mylopharyngodon piceus"                  </span></span>
<span><span class="co">#&gt; [32] "Nipponocypris sieboldii"                 </span></span>
<span><span class="co">#&gt; [33] "Nipponocypris temminckii"                </span></span>
<span><span class="co">#&gt; [34] "Opsariichthys platypus"                  </span></span>
<span><span class="co">#&gt; [35] "Opsariichthys uncirostris uncirostris"   </span></span>
<span><span class="co">#&gt; [36] "Oryzias latipes"                         </span></span>
<span><span class="co">#&gt; [37] "Plecoglossus altivelis altivelis"        </span></span>
<span><span class="co">#&gt; [38] "Pseudogobio spp."                        </span></span>
<span><span class="co">#&gt; [39] "Pseudorasbora parva"                     </span></span>
<span><span class="co">#&gt; [40] "Rhinogobius spp."                        </span></span>
<span><span class="co">#&gt; [41] "Rhodeus ocellatus ocellatus"             </span></span>
<span><span class="co">#&gt; [42] "Salangichthys microdon"                  </span></span>
<span><span class="co">#&gt; [43] "Sarcocheilichthys variegatus microoculus"</span></span>
<span><span class="co">#&gt; [44] "Silurus asotus"                          </span></span>
<span><span class="co">#&gt; [45] "Squalidus chankaensis biwae"             </span></span>
<span><span class="co">#&gt; [46] "Tachysurus tokiensis"                    </span></span>
<span><span class="co">#&gt; [47] "Tanakia lanceolata"                      </span></span>
<span><span class="co">#&gt; [48] "Tribolodon brandtii maruta"              </span></span>
<span><span class="co">#&gt; [49] "Tribolodon hakonensis"                   </span></span>
<span><span class="co">#&gt; [50] "Tridentiger spp."                        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $label$Sites</span></span>
<span><span class="co">#&gt;  [1] "1"  "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9"  "10" "11" "12" "13" "14" "15"</span></span>
<span><span class="co">#&gt; [16] "16" "17" "18" "19" "20" "21" "22" "23" "24" "25" "26" "27" "28" "29" "30"</span></span>
<span><span class="co">#&gt; [31] "31" "32" "33" "34" "35" "36" "37" "38" "39" "40" "41" "42" "43" "44" "45"</span></span>
<span><span class="co">#&gt; [46] "46" "47" "48" "49" "50"</span></span></code></pre></div>
<p>One can obtain posterior samples of <code>psi</code> with the
<code><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict()</a></code> function by specifying
<code>type = "samples"</code> instead of
<code>type = "quantiles"</code>.</p>
<p>For an example of specifying the <code>newdata</code> argument, let
us prepare a new dataset containing two sites with different
<code>riverbank</code> values. Recall that in <code>fit1</code>,
<code>psi</code> was expressed as a function of the site covariate
<code>riverbank</code>. Here, we use the new dataset created below to
predict <code>psi</code> of two sites with different
<code>riverbank</code> values. In the following, <code>fish_raw</code>
is a data object built into the package that stores the same data as
<code>fish</code> in the list format (see <code><a href="../reference/fish_raw.html">?fish_raw</a></code> for
documentation).</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">new_y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/array.html" class="external-link">array</a></span><span class="op">(</span><span class="fl">1</span>, dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">new_riverbank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">fish_raw</span><span class="op">$</span><span class="va">riverbank</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dimnames.html" class="external-link">dimnames</a></span><span class="op">(</span><span class="va">new_y</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dimnames.html" class="external-link">dimnames</a></span><span class="op">(</span><span class="va">fish_raw</span><span class="op">$</span><span class="va">y</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dimnames.html" class="external-link">dimnames</a></span><span class="op">(</span><span class="va">new_y</span><span class="op">)</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">fish_raw</span><span class="op">$</span><span class="va">riverbank</span><span class="op">)</span></span>
<span><span class="va">newdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/occumbData.html">occumbData</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">new_y</span>,</span>
<span>                      spec_cov <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>mismatch <span class="op">=</span> <span class="va">fish_raw</span><span class="op">$</span><span class="va">mismatch</span><span class="op">)</span>,</span>
<span>                      site_cov <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>riverbank <span class="op">=</span> <span class="va">new_riverbank</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Covariate values are relevant to the prediction but not the sequence
read counts. Hence, any hypothetical data may be given to the sequence
read count data, <code>y</code>, which was set to an array with elements
of 1, <code>new_y</code>, here. Let’s verify that <code>newdata</code>
is a dataset with <code>I = 50</code> species, <code>J = 2</code> sites,
and <code>K = 1</code> replicate, including <code>mismatch</code> and
<code>riverbank</code> as covariates. Note that the first site is
labeled (for demonstration purposes) as a site with vegetation
(<code>with_vegetation</code>) and the second site as a site without
vegetation (<code>without_vegetation</code>).</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">newdata</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Sequence read counts: </span></span>
<span><span class="co">#&gt;  Number of species, I = 50 </span></span>
<span><span class="co">#&gt;  Number of sites, J = 2 </span></span>
<span><span class="co">#&gt;  Maximum number of replicates per site, K = 1 </span></span>
<span><span class="co">#&gt;  Number of missing observations = 0 </span></span>
<span><span class="co">#&gt;  Number of replicates per site: 1 (average), 0 (sd) </span></span>
<span><span class="co">#&gt;  Sequencing depth: 50 (average), 0 (sd) </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Species covariates: </span></span>
<span><span class="co">#&gt;  mismatch (continuous) </span></span>
<span><span class="co">#&gt; Site covariates: </span></span>
<span><span class="co">#&gt;  riverbank (categorical) </span></span>
<span><span class="co">#&gt; Replicate covariates: </span></span>
<span><span class="co">#&gt;  (None) </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Labels for species: </span></span>
<span><span class="co">#&gt;  Abbottina rivularis, Acanthogobius lactipes, Acheilognathus macropterus, Acheilognathus rhombeus, Anguilla japonica, Biwia zezera, Carassius cuvieri, Carassius spp., Channa argus, Ctenopharyngodon idella, Cyprinus carpio, Gambusia affinis, Gnathopogon spp., Gymnogobius castaneus, Gymnogobius petschiliensis, Gymnogobius urotaenia, Hemibarbus spp., Hypomesus nipponensis, Hypophthalmichthys spp., Hyporhamphus intermedius, Ictalurus punctatus, Ischikauia steenackeri, Lepomis macrochirus macrochirus, Leucopsarion petersii, Megalobrama amblycephala, Micropterus dolomieu dolomieu, Micropterus salmoides, Misgurnus spp., Monopterus albus, Mugil cephalus cephalus, Mylopharyngodon piceus, Nipponocypris sieboldii, Nipponocypris temminckii, Opsariichthys platypus, Opsariichthys uncirostris uncirostris, Oryzias latipes, Plecoglossus altivelis altivelis, Pseudogobio spp., Pseudorasbora parva, Rhinogobius spp., Rhodeus ocellatus ocellatus, Salangichthys microdon, Sarcocheilichthys variegatus microoculus, Silurus asotus, Squalidus chankaensis biwae, Tachysurus tokiensis, Tanakia lanceolata, Tribolodon brandtii maruta, Tribolodon hakonensis, Tridentiger spp. </span></span>
<span><span class="co">#&gt; Labels for sites: </span></span>
<span><span class="co">#&gt;  with_vegetation, without_vegetation </span></span>
<span><span class="co">#&gt; Labels for replicates: </span></span>
<span><span class="co">#&gt;  (None) </span></span></code></pre></div>
<p>When the <code><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict()</a></code> function is applied with the
<code>newdata</code> argument, predictions are made using the covariate
values contained in the <code>newdata</code>. The following will provide
a prediction of <code>psi</code> for each of the sites with and without
vegetation.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">fit1</span>, newdata <span class="op">=</span> <span class="va">newdata</span>, parameter <span class="op">=</span> <span class="st">"psi"</span>, type <span class="op">=</span> <span class="st">"quantiles"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; , , 1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;            [,1]      [,2]      [,3]       [,4]       [,5]       [,6]      [,7]</span></span>
<span><span class="co">#&gt; 50%   0.6615133 0.2751231 0.3827295 0.15224665 0.17718177 0.09182312 0.8490621</span></span>
<span><span class="co">#&gt; 2.5%  0.4802434 0.1300821 0.2306797 0.05507663 0.06783645 0.02441254 0.7212899</span></span>
<span><span class="co">#&gt; 97.5% 0.7915599 0.4723714 0.5649520 0.32693805 0.38346795 0.24695832 0.9368627</span></span>
<span><span class="co">#&gt;            [,8]      [,9]     [,10]     [,11]      [,12]     [,13]      [,14]</span></span>
<span><span class="co">#&gt; 50%   0.9491461 0.6094722 0.4620570 0.9901477 0.08998769 0.8800989 0.15458012</span></span>
<span><span class="co">#&gt; 2.5%  0.8789348 0.4101236 0.2996333 0.9557121 0.02421783 0.7735010 0.05625485</span></span>
<span><span class="co">#&gt; 97.5% 0.9844834 0.8144525 0.6430711 0.9986149 0.23751947 0.9596028 0.31844988</span></span>
<span><span class="co">#&gt;            [,15]     [,16]     [,17]      [,18]     [,19]     [,20]     [,21]</span></span>
<span><span class="co">#&gt; 50%   0.12125019 0.9695970 0.9028737 0.15509196 0.5713331 0.2300419 0.8826944</span></span>
<span><span class="co">#&gt; 2.5%  0.03724413 0.9110775 0.8015870 0.05106154 0.3814354 0.1092197 0.7664569</span></span>
<span><span class="co">#&gt; 97.5% 0.26923379 0.9924366 0.9631803 0.34754250 0.7741282 0.4051336 0.9550034</span></span>
<span><span class="co">#&gt;           [,22]     [,23]      [,24]      [,25]      [,26]     [,27]     [,28]</span></span>
<span><span class="co">#&gt; 50%   0.7916404 0.7469745 0.06964821 0.15088247 0.10296476 0.8278437 0.9750893</span></span>
<span><span class="co">#&gt; 2.5%  0.6445037 0.5788305 0.01384897 0.04985234 0.02831969 0.7060083 0.9284540</span></span>
<span><span class="co">#&gt; 97.5% 0.9020019 0.8627692 0.19692563 0.34110667 0.25821933 0.9216507 0.9943395</span></span>
<span><span class="co">#&gt;            [,29]     [,30]      [,31]     [,32]     [,33]     [,34]     [,35]</span></span>
<span><span class="co">#&gt; 50%   0.06220973 0.4863689 0.07515230 0.4353972 0.2299977 0.8332137 0.5024979</span></span>
<span><span class="co">#&gt; 2.5%  0.01098996 0.2994199 0.01757379 0.2667359 0.1056246 0.6977411 0.3344813</span></span>
<span><span class="co">#&gt; 97.5% 0.19309849 0.6536841 0.21486234 0.6188191 0.4107509 0.9326166 0.6992168</span></span>
<span><span class="co">#&gt;           [,36]      [,37]     [,38]     [,39]     [,40]     [,41]      [,42]</span></span>
<span><span class="co">#&gt; 50%   0.5793313 0.09849281 0.5462387 0.9190994 0.9970209 0.8173368 0.09874564</span></span>
<span><span class="co">#&gt; 2.5%  0.4008518 0.02705771 0.3669093 0.8250025 0.9815437 0.6718077 0.02554623</span></span>
<span><span class="co">#&gt; 97.5% 0.7541354 0.25569027 0.7265439 0.9755158 0.9997452 0.9114977 0.26906733</span></span>
<span><span class="co">#&gt;            [,43]     [,44]     [,45]     [,46]      [,47]     [,48]     [,49]</span></span>
<span><span class="co">#&gt; 50%   0.15160773 0.5472967 0.3177699 0.1336518 0.16453410 0.3119199 0.1010661</span></span>
<span><span class="co">#&gt; 2.5%  0.05407187 0.3702903 0.1750710 0.0401887 0.06069858 0.1646373 0.0270361</span></span>
<span><span class="co">#&gt; 97.5% 0.32445164 0.7429405 0.4983112 0.3168030 0.32694679 0.4919569 0.2257502</span></span>
<span><span class="co">#&gt;           [,50]</span></span>
<span><span class="co">#&gt; 50%   0.9885246</span></span>
<span><span class="co">#&gt; 2.5%  0.9521853</span></span>
<span><span class="co">#&gt; 97.5% 0.9984315</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; , , 2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;            [,1]       [,2]       [,3]       [,4]       [,5]       [,6]</span></span>
<span><span class="co">#&gt; 50%   0.4756873 0.16831019 0.17344502 0.06827157 0.08065680 0.03799725</span></span>
<span><span class="co">#&gt; 2.5%  0.3210251 0.07468112 0.08268459 0.02152790 0.02558621 0.00904520</span></span>
<span><span class="co">#&gt; 97.5% 0.6477671 0.31803370 0.30375314 0.16653407 0.20628815 0.10914663</span></span>
<span><span class="co">#&gt;            [,7]      [,8]      [,9]     [,10]     [,11]      [,12]     [,13]</span></span>
<span><span class="co">#&gt; 50%   0.5747380 0.8277408 0.4451700 0.2415741 0.9426904 0.04637033 0.6435848</span></span>
<span><span class="co">#&gt; 2.5%  0.4085796 0.6832309 0.2727587 0.1336498 0.8361653 0.01254011 0.4719155</span></span>
<span><span class="co">#&gt; 97.5% 0.7316505 0.9241055 0.6653103 0.3976390 0.9855532 0.12900086 0.7834428</span></span>
<span><span class="co">#&gt;            [,14]      [,15]     [,16]     [,17]      [,18]     [,19]     [,20]</span></span>
<span><span class="co">#&gt; 50%   0.07037047 0.06163691 0.8602899 0.6909106 0.08289891 0.3366483 0.1156426</span></span>
<span><span class="co">#&gt; 2.5%  0.02364689 0.01866216 0.7192849 0.5273816 0.02726444 0.2012909 0.0498471</span></span>
<span><span class="co">#&gt; 97.5% 0.16850684 0.15171748 0.9468762 0.8219908 0.20203910 0.5227359 0.2284146</span></span>
<span><span class="co">#&gt;           [,21]     [,22]     [,23]       [,24]      [,25]      [,26]     [,27]</span></span>
<span><span class="co">#&gt; 50%   0.7204299 0.5831452 0.5653456 0.030925710 0.06781351 0.05881219 0.6000844</span></span>
<span><span class="co">#&gt; 2.5%  0.5574411 0.4218203 0.4080076 0.006236538 0.02063876 0.01733949 0.4410841</span></span>
<span><span class="co">#&gt; 97.5% 0.8503492 0.7422631 0.7276702 0.094870829 0.17049638 0.16207666 0.7470483</span></span>
<span><span class="co">#&gt;           [,28]       [,29]     [,30]      [,31]      [,32]      [,33]</span></span>
<span><span class="co">#&gt; 50%   0.9126102 0.035225441 0.3407391 0.04236923 0.16616006 0.08886842</span></span>
<span><span class="co">#&gt; 2.5%  0.8020100 0.007672009 0.2031182 0.01018349 0.07272355 0.03188771</span></span>
<span><span class="co">#&gt; 97.5% 0.9696633 0.110552419 0.5178231 0.13816851 0.29161845 0.18440940</span></span>
<span><span class="co">#&gt;           [,34]     [,35]     [,36]      [,37]     [,38]     [,39]     [,40]</span></span>
<span><span class="co">#&gt; 50%   0.5159626 0.2344728 0.3705463 0.04820736 0.2161039 0.7693729 0.9831972</span></span>
<span><span class="co">#&gt; 2.5%  0.3512136 0.1224432 0.2295261 0.01206818 0.1039605 0.6147098 0.9306534</span></span>
<span><span class="co">#&gt; 97.5% 0.6751210 0.3846941 0.5527593 0.13468654 0.3594184 0.8923270 0.9976299</span></span>
<span><span class="co">#&gt;           [,41]      [,42]      [,43]     [,44]      [,45]      [,46]</span></span>
<span><span class="co">#&gt; 50%   0.6527351 0.06006526 0.06753850 0.2371226 0.15535572 0.05145973</span></span>
<span><span class="co">#&gt; 2.5%  0.4967540 0.01537025 0.02157825 0.1134572 0.07282467 0.01382803</span></span>
<span><span class="co">#&gt; 97.5% 0.7956659 0.16542028 0.16515022 0.3848913 0.27405126 0.13368309</span></span>
<span><span class="co">#&gt;            [,47]      [,48]      [,49]     [,50]</span></span>
<span><span class="co">#&gt; 50%   0.08990812 0.13653424 0.05038579 0.9428926</span></span>
<span><span class="co">#&gt; 2.5%  0.03330197 0.05777936 0.01466287 0.8401821</span></span>
<span><span class="co">#&gt; 97.5% 0.20500919 0.25327092 0.12890174 0.9873862</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attr(,"parameter")</span></span>
<span><span class="co">#&gt; [1] "psi"</span></span>
<span><span class="co">#&gt; attr(,"scale")</span></span>
<span><span class="co">#&gt; [1] "response"</span></span>
<span><span class="co">#&gt; attr(,"dimension")</span></span>
<span><span class="co">#&gt; [1] "Statistics" "Species"    "Sites"     </span></span>
<span><span class="co">#&gt; attr(,"label")</span></span>
<span><span class="co">#&gt; attr(,"label")$Statistics</span></span>
<span><span class="co">#&gt; [1] "50%"   "2.5%"  "97.5%"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attr(,"label")$Species</span></span>
<span><span class="co">#&gt;  [1] "Abbottina rivularis"                     </span></span>
<span><span class="co">#&gt;  [2] "Acanthogobius lactipes"                  </span></span>
<span><span class="co">#&gt;  [3] "Acheilognathus macropterus"              </span></span>
<span><span class="co">#&gt;  [4] "Acheilognathus rhombeus"                 </span></span>
<span><span class="co">#&gt;  [5] "Anguilla japonica"                       </span></span>
<span><span class="co">#&gt;  [6] "Biwia zezera"                            </span></span>
<span><span class="co">#&gt;  [7] "Carassius cuvieri"                       </span></span>
<span><span class="co">#&gt;  [8] "Carassius spp."                          </span></span>
<span><span class="co">#&gt;  [9] "Channa argus"                            </span></span>
<span><span class="co">#&gt; [10] "Ctenopharyngodon idella"                 </span></span>
<span><span class="co">#&gt; [11] "Cyprinus carpio"                         </span></span>
<span><span class="co">#&gt; [12] "Gambusia affinis"                        </span></span>
<span><span class="co">#&gt; [13] "Gnathopogon spp."                        </span></span>
<span><span class="co">#&gt; [14] "Gymnogobius castaneus"                   </span></span>
<span><span class="co">#&gt; [15] "Gymnogobius petschiliensis"              </span></span>
<span><span class="co">#&gt; [16] "Gymnogobius urotaenia"                   </span></span>
<span><span class="co">#&gt; [17] "Hemibarbus spp."                         </span></span>
<span><span class="co">#&gt; [18] "Hypomesus nipponensis"                   </span></span>
<span><span class="co">#&gt; [19] "Hypophthalmichthys spp."                 </span></span>
<span><span class="co">#&gt; [20] "Hyporhamphus intermedius"                </span></span>
<span><span class="co">#&gt; [21] "Ictalurus punctatus"                     </span></span>
<span><span class="co">#&gt; [22] "Ischikauia steenackeri"                  </span></span>
<span><span class="co">#&gt; [23] "Lepomis macrochirus macrochirus"         </span></span>
<span><span class="co">#&gt; [24] "Leucopsarion petersii"                   </span></span>
<span><span class="co">#&gt; [25] "Megalobrama amblycephala"                </span></span>
<span><span class="co">#&gt; [26] "Micropterus dolomieu dolomieu"           </span></span>
<span><span class="co">#&gt; [27] "Micropterus salmoides"                   </span></span>
<span><span class="co">#&gt; [28] "Misgurnus spp."                          </span></span>
<span><span class="co">#&gt; [29] "Monopterus albus"                        </span></span>
<span><span class="co">#&gt; [30] "Mugil cephalus cephalus"                 </span></span>
<span><span class="co">#&gt; [31] "Mylopharyngodon piceus"                  </span></span>
<span><span class="co">#&gt; [32] "Nipponocypris sieboldii"                 </span></span>
<span><span class="co">#&gt; [33] "Nipponocypris temminckii"                </span></span>
<span><span class="co">#&gt; [34] "Opsariichthys platypus"                  </span></span>
<span><span class="co">#&gt; [35] "Opsariichthys uncirostris uncirostris"   </span></span>
<span><span class="co">#&gt; [36] "Oryzias latipes"                         </span></span>
<span><span class="co">#&gt; [37] "Plecoglossus altivelis altivelis"        </span></span>
<span><span class="co">#&gt; [38] "Pseudogobio spp."                        </span></span>
<span><span class="co">#&gt; [39] "Pseudorasbora parva"                     </span></span>
<span><span class="co">#&gt; [40] "Rhinogobius spp."                        </span></span>
<span><span class="co">#&gt; [41] "Rhodeus ocellatus ocellatus"             </span></span>
<span><span class="co">#&gt; [42] "Salangichthys microdon"                  </span></span>
<span><span class="co">#&gt; [43] "Sarcocheilichthys variegatus microoculus"</span></span>
<span><span class="co">#&gt; [44] "Silurus asotus"                          </span></span>
<span><span class="co">#&gt; [45] "Squalidus chankaensis biwae"             </span></span>
<span><span class="co">#&gt; [46] "Tachysurus tokiensis"                    </span></span>
<span><span class="co">#&gt; [47] "Tanakia lanceolata"                      </span></span>
<span><span class="co">#&gt; [48] "Tribolodon brandtii maruta"              </span></span>
<span><span class="co">#&gt; [49] "Tribolodon hakonensis"                   </span></span>
<span><span class="co">#&gt; [50] "Tridentiger spp."                        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attr(,"label")$Sites</span></span>
<span><span class="co">#&gt; [1] "with_vegetation"    "without_vegetation"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="assess-goodness-of-fit">Assess goodness-of-fit<a class="anchor" aria-label="anchor" href="#assess-goodness-of-fit"></a>
</h2>
<p>Assessing the goodness-of-fit of a model is an essential step in data
analysis based on statistical models. The goodness-of-fit of the models
fitted with <code><a href="../reference/occumb.html">occumb()</a></code> is evaluated using the
<code><a href="../reference/gof.html">gof()</a></code> function, which computes the Bayesian p-value using
the posterior predictive check approach. The Bayesian p-value may take
extreme values (e.g., <code>p &lt; 0.05</code> or
<code>p &gt; 0.95</code>) when the model was poorly fitted.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gof_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gof.html">gof</a></span><span class="op">(</span><span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p>For faster computation, it is recommended that the <code>cores</code>
argument be specified explicitly for parallel computations. By default,
the <code><a href="../reference/gof.html">gof()</a></code> function outputs a scatter plot of fit statistics
(in this case, the default Freeman-Tukey statistics).</p>
<p>Enter the name of the resulting object to summarize the results.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gof_result</span></span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Posterior predictive check for an occumbFit object:</span></span>
<span><span class="co">#&gt;  Statistics: Freeman-Tukey </span></span>
<span><span class="co">#&gt;  p-value:    0.2005 </span></span>
<span><span class="co">#&gt;  Discrepancy statistics for observed data:   542.81 (mean), 17.92 (sd) </span></span>
<span><span class="co">#&gt;  Discrepancy statistics for replicated data: 523.28 (mean), 16.47 (sd) </span></span></code></pre></div>
<p>A moderate p-value indicated a lack of evidence of the model’s
inadequacy in fitting.</p>
<p>The <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> function can be used to (re)draw a scatter
plot of the fit statistics for the resulting object.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gof_result</span><span class="op">)</span></span></code></pre></div>
<p><img src="images%2Fgof_fit1.png"></p>
</div>
<div class="section level2">
<h2 id="analyze-study-design">Analyze study design<a class="anchor" aria-label="anchor" href="#analyze-study-design"></a>
</h2>
<p>The model fitted with <code><a href="../reference/occumb.html">occumb()</a></code> can be used to identify
eDNA metabarcoding study designs that can effectively detect the species
present. How many sites, within-site replicates, and sequencing depths
are required for reliable species detection? What is the best balance
between the number of sites visited, number of within-site replicates,
and sequencing depth under a limited budget? The multispecies site
occupancy model fit by <code><a href="../reference/occumb.html">occumb()</a></code> answers these questions by
predicting the number of species expected to be detected using specific
study designs.</p>
<p><code><a href="../reference/eval_util_L.html">eval_util_L()</a></code> and <code><a href="../reference/eval_util_R.html">eval_util_R()</a></code> are
functions available for these purposes. These two functions assume
species diversity assessments at different spatial scales (i.e., the
former is for local (L) and the latter for regional (R)). Specifically,
<code><a href="../reference/eval_util_L.html">eval_util_L()</a></code> is appropriate if you are interested in
assessing species diversity only for the study sites included in the
dataset, and <code><a href="../reference/eval_util_R.html">eval_util_R()</a></code> is appropriate if you are
interested in assessing species diversity in a broader area that
includes the study sites in the dataset (i.e., the population of sites
or a “metacommunity”).</p>
<p>In the context of the <code>fish</code> dataset,
<code><a href="../reference/eval_util_L.html">eval_util_L()</a></code> evaluates the expected number of species
detected per site at 50 sites in the dataset under different
combinations of number of replicates and sequencing depth. For example,
the following will give the expected number of species detected per site
(referred to as <code>Utility</code> below) when the number of
replicates <code>K</code> and the sequencing depth <code>N</code> take
the values (1, 2, 3) and (1000, 10000, 100000), respectively.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">utilL1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/eval_util_L.html">eval_util_L</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/expand.grid.html" class="external-link">expand.grid</a></span><span class="op">(</span>K <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, N <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1E3</span>, <span class="fl">1E4</span>, <span class="fl">1E5</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                      <span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilL1</span></span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;   K     N  Utility</span></span>
<span><span class="co">#&gt; 1 1 1e+03 13.03225</span></span>
<span><span class="co">#&gt; 2 2 1e+03 16.72177</span></span>
<span><span class="co">#&gt; 3 3 1e+03 18.21783</span></span>
<span><span class="co">#&gt; 4 1 1e+04 14.34826</span></span>
<span><span class="co">#&gt; 5 2 1e+04 17.72304</span></span>
<span><span class="co">#&gt; 6 3 1e+04 18.92009</span></span>
<span><span class="co">#&gt; 7 1 1e+05 15.04591</span></span>
<span><span class="co">#&gt; 8 2 1e+05 18.20935</span></span>
<span><span class="co">#&gt; 9 3 1e+05 19.23478</span></span></code></pre></div>
<p>It can be seen that as <code>K</code> and <code>N</code> increase,
the <code>Utility</code> value increases. For faster computation, it is
recommended that the <code>cores</code> argument be explicitly specified
for parallel computation.</p>
<p>If one knows the cost per sequence read for high-throughput
sequencing, cost per replicate for library preparation, and research
budget values, the <code><a href="../reference/list_cond_L.html">list_cond_L()</a></code> function can be used to
obtain a set of feasible settings under these cost and budget values.
This can be applied to <code><a href="../reference/eval_util_L.html">eval_util_L()</a></code> to identify the
optimal study design under the budget constraints.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">settings</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/list_cond_L.html">list_cond_L</a></span><span class="op">(</span>budget <span class="op">=</span> <span class="fl">875</span> <span class="op">*</span> <span class="fl">1E3</span>,</span>
<span>                        lambda1 <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                        lambda2 <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>                        <span class="va">fit1</span><span class="op">)</span></span>
<span><span class="va">utilL2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/eval_util_L.html">eval_util_L</a></span><span class="op">(</span><span class="va">settings</span>, <span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilL2</span></span></code></pre></div>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;   budget lambda1 lambda2 K          N  Utility</span></span>
<span><span class="co">#&gt; 1 875000    0.01    5000 1 1250000.00 15.45671</span></span>
<span><span class="co">#&gt; 2 875000    0.01    5000 2  375000.00 18.37862</span></span>
<span><span class="co">#&gt; 3 875000    0.01    5000 3   83333.33 19.21953</span></span></code></pre></div>
<p>Hence, under this specific budget and cost, it is best to have
<code>K = 3</code> replicates per site for effective species
detection.</p>
<p>In contrast to <code><a href="../reference/eval_util_L.html">eval_util_L()</a></code>, <code><a href="../reference/eval_util_R.html">eval_util_R()</a></code>
considers differences in the number of sites visited. Nevertheless,
<code><a href="../reference/eval_util_R.html">eval_util_R()</a></code> can be applied similarly as
<code><a href="../reference/eval_util_L.html">eval_util_L()</a></code>. In <code><a href="../reference/eval_util_R.html">eval_util_R()</a></code>,
<code>Utility</code> is the number of species expected to be detected in
the region of interest under the settings of number of sites
<code>J</code>, number of replicates <code>K</code>, and sequencing
depth <code>N</code>. Because <code><a href="../reference/list_cond_R.html">list_cond_R()</a></code> may return a
large list of possible settings, one may want to obtain a manually
restricted list using its <code>J</code> and/or <code>K</code>
arguments.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">utilR1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/eval_util_R.html">eval_util_R</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/expand.grid.html" class="external-link">expand.grid</a></span><span class="op">(</span>J <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, K <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, N <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1E3</span>, <span class="fl">1E4</span>, <span class="fl">1E5</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                      <span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilR1</span></span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; 1  1 1 1e+03 13.06227</span></span>
<span><span class="co">#&gt; 2  2 1 1e+03 18.77537</span></span>
<span><span class="co">#&gt; 3  3 1 1e+03 22.21901</span></span>
<span><span class="co">#&gt; 4  1 2 1e+03 16.75208</span></span>
<span><span class="co">#&gt; 5  2 2 1e+03 22.80510</span></span>
<span><span class="co">#&gt; 6  3 2 1e+03 26.33776</span></span>
<span><span class="co">#&gt; 7  1 3 1e+03 18.18526</span></span>
<span><span class="co">#&gt; 8  2 3 1e+03 24.63089</span></span>
<span><span class="co">#&gt; 9  3 3 1e+03 28.13433</span></span>
<span><span class="co">#&gt; 10 1 1 1e+04 14.36287</span></span>
<span><span class="co">#&gt; 11 2 1 1e+04 20.30642</span></span>
<span><span class="co">#&gt; 12 3 1 1e+04 23.85725</span></span>
<span><span class="co">#&gt; 13 1 2 1e+04 17.65318</span></span>
<span><span class="co">#&gt; 14 2 2 1e+04 23.98433</span></span>
<span><span class="co">#&gt; 15 3 2 1e+04 27.55905</span></span>
<span><span class="co">#&gt; 16 1 3 1e+04 19.03370</span></span>
<span><span class="co">#&gt; 17 2 3 1e+04 25.30884</span></span>
<span><span class="co">#&gt; 18 3 3 1e+04 29.14037</span></span>
<span><span class="co">#&gt; 19 1 1 1e+05 14.97290</span></span>
<span><span class="co">#&gt; 20 2 1 1e+05 21.16216</span></span>
<span><span class="co">#&gt; 21 3 1 1e+05 24.76543</span></span>
<span><span class="co">#&gt; 22 1 2 1e+05 18.25193</span></span>
<span><span class="co">#&gt; 23 2 2 1e+05 24.64478</span></span>
<span><span class="co">#&gt; 24 3 2 1e+05 28.20150</span></span>
<span><span class="co">#&gt; 25 1 3 1e+05 19.33757</span></span>
<span><span class="co">#&gt; 26 2 3 1e+05 25.85813</span></span>
<span><span class="co">#&gt; 27 3 3 1e+05 29.68874</span></span></code></pre></div>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">settings</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/list_cond_R.html">list_cond_R</a></span><span class="op">(</span>budget <span class="op">=</span> <span class="fl">1125</span> <span class="op">*</span> <span class="fl">1E3</span>,</span>
<span>                        lambda1 <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                        lambda2 <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>                        lambda3 <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>                        J <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">50</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>                        K <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilR2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/eval_util_R.html">eval_util_R</a></span><span class="op">(</span><span class="va">settings</span>, <span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilR2</span></span></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;     budget lambda1 lambda2 lambda3  J K           N  Utility</span></span>
<span><span class="co">#&gt; 1  1125000    0.01    5000    5000  5 1 21500000.00 30.04136</span></span>
<span><span class="co">#&gt; 2  1125000    0.01    5000    5000 10 1 10250000.00 35.64544</span></span>
<span><span class="co">#&gt; 3  1125000    0.01    5000    5000 15 1  6500000.00 38.60797</span></span>
<span><span class="co">#&gt; 4  1125000    0.01    5000    5000 20 1  4625000.00 40.60944</span></span>
<span><span class="co">#&gt; 5  1125000    0.01    5000    5000 25 1  3500000.00 42.02112</span></span>
<span><span class="co">#&gt; 6  1125000    0.01    5000    5000 30 1  2750000.00 43.15096</span></span>
<span><span class="co">#&gt; 7  1125000    0.01    5000    5000 35 1  2214285.71 43.89833</span></span>
<span><span class="co">#&gt; 8  1125000    0.01    5000    5000 40 1  1812500.00 44.71556</span></span>
<span><span class="co">#&gt; 9  1125000    0.01    5000    5000 45 1  1500000.00 45.22666</span></span>
<span><span class="co">#&gt; 10 1125000    0.01    5000    5000 50 1  1250000.00 45.71167</span></span>
<span><span class="co">#&gt; 11 1125000    0.01    5000    5000  5 2 10500000.00 33.34409</span></span>
<span><span class="co">#&gt; 12 1125000    0.01    5000    5000 10 2  4875000.00 38.81906</span></span>
<span><span class="co">#&gt; 13 1125000    0.01    5000    5000 15 2  3000000.00 41.58322</span></span>
<span><span class="co">#&gt; 14 1125000    0.01    5000    5000 20 2  2062500.00 43.47530</span></span>
<span><span class="co">#&gt; 15 1125000    0.01    5000    5000 25 2  1500000.00 44.69525</span></span>
<span><span class="co">#&gt; 16 1125000    0.01    5000    5000 30 2  1125000.00 45.58398</span></span>
<span><span class="co">#&gt; 17 1125000    0.01    5000    5000 35 2   857142.86 46.27036</span></span>
<span><span class="co">#&gt; 18 1125000    0.01    5000    5000 40 2   656250.00 46.82490</span></span>
<span><span class="co">#&gt; 19 1125000    0.01    5000    5000 45 2   500000.00 47.24841</span></span>
<span><span class="co">#&gt; 20 1125000    0.01    5000    5000 50 2   375000.00 47.56585</span></span>
<span><span class="co">#&gt; 21 1125000    0.01    5000    5000  5 3  6833333.33 34.55557</span></span>
<span><span class="co">#&gt; 22 1125000    0.01    5000    5000 10 3  3083333.33 40.03808</span></span>
<span><span class="co">#&gt; 23 1125000    0.01    5000    5000 15 3  1833333.33 42.85560</span></span>
<span><span class="co">#&gt; 24 1125000    0.01    5000    5000 20 3  1208333.33 44.51759</span></span>
<span><span class="co">#&gt; 25 1125000    0.01    5000    5000 25 3   833333.33 45.67757</span></span>
<span><span class="co">#&gt; 26 1125000    0.01    5000    5000 30 3   583333.33 46.48685</span></span>
<span><span class="co">#&gt; 27 1125000    0.01    5000    5000 35 3   404761.90 47.14072</span></span>
<span><span class="co">#&gt; 28 1125000    0.01    5000    5000 40 3   270833.33 47.60570</span></span>
<span><span class="co">#&gt; 29 1125000    0.01    5000    5000 45 3   166666.67 47.93433</span></span>
<span><span class="co">#&gt; 30 1125000    0.01    5000    5000 50 3    83333.33 48.12801</span></span>
<span><span class="co">#&gt; 31 1125000    0.01    5000    5000  5 4  5000000.00 35.17767</span></span>
<span><span class="co">#&gt; 32 1125000    0.01    5000    5000 10 4  2187500.00 40.72769</span></span>
<span><span class="co">#&gt; 33 1125000    0.01    5000    5000 15 4  1250000.00 43.51095</span></span>
<span><span class="co">#&gt; 34 1125000    0.01    5000    5000 20 4   781250.00 45.16898</span></span>
<span><span class="co">#&gt; 35 1125000    0.01    5000    5000 25 4   500000.00 46.23139</span></span>
<span><span class="co">#&gt; 36 1125000    0.01    5000    5000 30 4   312500.00 46.98586</span></span>
<span><span class="co">#&gt; 37 1125000    0.01    5000    5000 35 4   178571.43 47.53393</span></span>
<span><span class="co">#&gt; 38 1125000    0.01    5000    5000 40 4    78125.00 47.89744</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Keiichi Fukaya, Mika Takahashi.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
