<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="occumb">
<title>Introduction to occumb • occumb</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to occumb">
<meta property="og:description" content="occumb">
<meta name="robots" content="noindex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">occumb</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.0.1.9002</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/occumb.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/model_specification.html">Model specification in occumb()</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/fukayak/occumb/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to occumb</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/fukayak/occumb/blob/HEAD/vignettes/occumb.Rmd" class="external-link"><code>vignettes/occumb.Rmd</code></a></small>
      <div class="d-none name"><code>occumb.Rmd</code></div>
    </div>

    
    
<p>This section briefly describes the steps involved in a series of
model-based analyses of eDNA metabarcoding data that can be performed
using the <code>occumb</code> R package.</p>
<div class="section level2">
<h2 id="prepare-dataset">Prepare dataset<a class="anchor" aria-label="anchor" href="#prepare-dataset"></a>
</h2>
<p>We will use the package’s built-in data <code>fish</code> (see
<code><a href="../reference/fish.html">?fish</a></code> for documentation) to see how to analyze sequence
count data with the <code>occumb</code> package. The
<code><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary()</a></code> function can be used to display an overview of
the dataset.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">fish</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">fish</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Sequence read counts: </span></span>
<span><span class="co">#&gt;  Number of species, I = 50 </span></span>
<span><span class="co">#&gt;  Number of sites, J = 50 </span></span>
<span><span class="co">#&gt;  Maximum number of replicates per site, K = 3 </span></span>
<span><span class="co">#&gt;  Number of missing observations = 6 </span></span>
<span><span class="co">#&gt;  Number of replicates per site: 2.88 (average), 0.33 (sd)</span></span>
<span><span class="co">#&gt;  Sequencing depth: 77910 (average), 98034.7 (sd)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Species covariates: </span></span>
<span><span class="co">#&gt;  mismatch (continuous) </span></span>
<span><span class="co">#&gt; Site covariates: </span></span>
<span><span class="co">#&gt;  riverbank (categorical) </span></span>
<span><span class="co">#&gt; Replicate covariates: </span></span>
<span><span class="co">#&gt;  (None) </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Labels for species: </span></span>
<span><span class="co">#&gt;  Abbottina rivularis, Acanthogobius lactipes, Acheilognathus macropterus, Acheilognathus rhombeus, Anguilla japonica, Biwia zezera, Carassius cuvieri, Carassius spp., Channa argus, Ctenopharyngodon idella, Cyprinus carpio, Gambusia affinis, Gnathopogon spp., Gymnogobius castaneus, Gymnogobius petschiliensis, Gymnogobius urotaenia, Hemibarbus spp., Hypomesus nipponensis, Hypophthalmichthys spp., Hyporhamphus intermedius, Ictalurus punctatus, Ischikauia steenackeri, Lepomis macrochirus macrochirus, Leucopsarion petersii, Megalobrama amblycephala, Micropterus dolomieu dolomieu, Micropterus salmoides, Misgurnus spp., Monopterus albus, Mugil cephalus cephalus, Mylopharyngodon piceus, Nipponocypris sieboldii, Nipponocypris temminckii, Opsariichthys platypus, Opsariichthys uncirostris uncirostris, Oryzias latipes, Plecoglossus altivelis altivelis, Pseudogobio spp., Pseudorasbora parva, Rhinogobius spp., Rhodeus ocellatus ocellatus, Salangichthys microdon, Sarcocheilichthys variegatus microoculus, Silurus asotus, Squalidus chankaensis biwae, Tachysurus tokiensis, Tanakia lanceolata, Tribolodon brandtii maruta, Tribolodon hakonensis, Tridentiger spp. </span></span>
<span><span class="co">#&gt; Labels for sites: </span></span>
<span><span class="co">#&gt;  1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50 </span></span>
<span><span class="co">#&gt; Labels for replicates: </span></span>
<span><span class="co">#&gt;  L, C, R </span></span></code></pre></div>
<p>Note that a summary of the sequence read count data is found in the
first block of the output. It shows that the data was obtained from
samples of 50 sites <span class="math inline">\(\times\)</span> (up to)
3 replicates, with 50 fish species recorded. A summary of the missing
samples, number of replicates per site, and sequencing depth (i.e., the
total number of sequence reads per sample) are also presented. The
second block of the output indicates that <code>fish</code> data also
has two covariates, one continuous species covariate,
<code>mismatch</code>, and one discrete site covariate,
<code>riverbank</code>, which can be used in the following analysis. The
third block of the output shows labels given to the 50 species, 50
sites, and three replicates.</p>
<p>As such, the <code>fish</code> data brings together a set of data
relevant to the analysis with the <code>occumb</code> package. When you
are analyzing your own dataset, you can use <code><a href="../reference/occumbData.html">occumbData()</a></code> to
set up your data object.</p>
<p>If you want to display the entire data, just type
<code>fish</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fish</span> <span class="co"># Result not shown</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fit-models">Fit models<a class="anchor" aria-label="anchor" href="#fit-models"></a>
</h2>
<p>With the <code>fish</code> dataset (or your dataset constructed with
<code><a href="../reference/occumbData.html">occumbData()</a></code>), you can fit a multispecies site occupancy
model for eDNA metabarcoding using the <code><a href="../reference/occumb.html">occumb()</a></code> function
which fits the model in a fully Bayesian approach via the Markov Chain
Monte Carlo (MCMC) method implemented in <a href="https://mcmc-jags.sourceforge.io/" class="external-link">JAGS</a>.</p>
<p><code><a href="../reference/occumb.html">occumb()</a></code> has several arguments that accept model
formulas to fit different model variants using the usual R formula
syntax. See <code><a href="../articles/model_specification.html">vignette("model_specification")</a></code> for an overview
of the models that <code><a href="../reference/occumb.html">occumb()</a></code> can fit and details on how to
specify models.</p>
<p>Fitting a null model (i.e., an intercept-only model, though the value
of the intercept is species-specific) is very simple, as follows.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit0</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/occumb.html">occumb</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">fish</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Setting <code>parallel = TRUE</code> is recommended for faster model
fitting through parallel computation. A model incorporating species and
site covariates in the <code>fish</code> dataset can be fitted, for
example, as follows.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/occumb.html">occumb</a></span><span class="op">(</span>formula_psi <span class="op">=</span> <span class="op">~</span> <span class="va">riverbank</span>,</span>
<span>               formula_phi_shared <span class="op">=</span> <span class="op">~</span> <span class="va">mismatch</span>,</span>
<span>               data <span class="op">=</span> <span class="va">fish</span>,</span>
<span>               parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Here, <code>riverbank</code> is specified as the covariate for site
occupancy probability (<code>psi</code>) in the <code>formula_psi</code>
argument and <code>mismatch</code> as the covariate for relative
dominance of sequence (<code>phi</code>) in the
<code>formula_phi_shared</code> argument. In the latter, since
<code>mismatch</code> is a species covariate, the
<code>formula_phi_shared</code> argument is used instead of
<code>formula_phi</code> so that the effect of <code>mismatch</code> is
constant across species; see
<code><a href="../articles/model_specification.html">vignette("model_specification")</a></code> for more details.</p>
<p>The above two model fits can take approximately one hour. Given that
<code><a href="../reference/occumb.html">occumb()</a></code> fits a class of complex hierarchical models with
many parameters and unknown variables, it may require, depending on the
dataset size, a long time and/or a lot of memory for model fitting.</p>
<p><code><a href="../reference/occumb.html">occumb()</a></code> also has several arguments to control MCMC
computation for fitting the model. For example, to obtain more precise
posterior estimates than the example above, you can explicitly set the
<code>n.iter</code> and <code>n.thin</code> arguments to get longer and
less autocorrelated MCMC samples.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit1x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/occumb.html">occumb</a></span><span class="op">(</span>formula_psi <span class="op">=</span> <span class="op">~</span> <span class="va">riverbank</span>,</span>
<span>                formula_phi_shared <span class="op">=</span> <span class="op">~</span> <span class="va">mismatch</span>,</span>
<span>                data <span class="op">=</span> <span class="va">fish</span>,</span>
<span>                n.thin <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                n.iter <span class="op">=</span> <span class="fl">40000</span>,</span>
<span>                parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="check-posterior-samples">Check posterior samples<a class="anchor" aria-label="anchor" href="#check-posterior-samples"></a>
</h2>
<p>With the <code>fit1</code> object, we will look at how to access and
diagnose the posterior sample obtained using <code><a href="../reference/occumb.html">occumb()</a></code>.</p>
<p>Just enter the object name <code>fit1</code> displays a summary of
the MCMC results, including a table of posterior estimates for each
saved parameter.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit1</span> <span class="co"># Result not shown</span></span></code></pre></div>
<p>The <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> function draws a trace plot and a density
plot for each parameter.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit1</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/plot_fit1.png"></p>
<p>Users who have used the <a href="https://github.com/kenkellner/jagsUI" class="external-link">jagsUI</a> package should be
familiar with the output of these operations. Model fitting in
<code><a href="../reference/occumb.html">occumb()</a></code> relies on functions provided by the
<a href="https://github.com/kenkellner/jagsUI" class="external-link">jagsUI</a> package, and the above operations respectively
apply the print and plot methods for <a href="https://github.com/kenkellner/jagsUI" class="external-link">jagsUI</a> objects.</p>
<p>The <code><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary()</a></code> function provides a high-level summary of
the result. This is based on the summary method for
<a href="https://github.com/kenkellner/jagsUI" class="external-link">jagsUI</a> objects and modified for the <code><a href="../reference/occumb.html">occumb()</a></code>
output, which displays data summary, model equations, MCMC settings, and
a quick overview of the convergence of relevant model parameters.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">fit1</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Summary for an occumbFit object </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Summary of data:</span></span>
<span><span class="co">#&gt;  Number of species, I = 50 </span></span>
<span><span class="co">#&gt;  Number of sites, J = 50 </span></span>
<span><span class="co">#&gt;  Maximum number of replicates per site, K = 3 </span></span>
<span><span class="co">#&gt;  Number of missing observations = 6 </span></span>
<span><span class="co">#&gt;  Number of replicates per site: 2.88 (average), 0.33 (sd) </span></span>
<span><span class="co">#&gt;  Sequencing depth: 77910 (average), 98034.7 (sd) </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Model specification:</span></span>
<span><span class="co">#&gt;  formula_phi:          ~ 1 </span></span>
<span><span class="co">#&gt;  formula_theta:        ~ 1 </span></span>
<span><span class="co">#&gt;  formula_psi:          ~ riverbank </span></span>
<span><span class="co">#&gt;  formula_phi_shared:   ~ mismatch </span></span>
<span><span class="co">#&gt;  formula_theta_shared: ~ 1 </span></span>
<span><span class="co">#&gt;  formula_psi_shared:   ~ 1 </span></span>
<span><span class="co">#&gt;  prior_prec:           1e-04 </span></span>
<span><span class="co">#&gt;  prior_ulim:           10000 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Saved parameters:</span></span>
<span><span class="co">#&gt;  Mu sigma rho alpha beta gamma alpha_shared phi theta psi z pi deviance </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MCMC ran in parallel for 43.593 minutes at time 2023-09-11 20:16:31.787026:</span></span>
<span><span class="co">#&gt;  For each of 4 chains:</span></span>
<span><span class="co">#&gt;   Adaptation:            100 iterations (sufficient)</span></span>
<span><span class="co">#&gt;   Burn-in:               10000 iterations</span></span>
<span><span class="co">#&gt;   Thin rate:             10 iterations</span></span>
<span><span class="co">#&gt;   Total chain length:    20100 iterations</span></span>
<span><span class="co">#&gt;   Posterior sample size: 1000 draws</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Summary of posterior samples: </span></span>
<span><span class="co">#&gt;  Mu: </span></span>
<span><span class="co">#&gt;   Number of parameters: 4 </span></span>
<span><span class="co">#&gt;   Rhat:  1.009 (min), 1.027 (median), 1.043 (mean), 1.109 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 30 (min), 117.5 (median), 154.2 (mean), 352 (max) </span></span>
<span><span class="co">#&gt;  sigma: </span></span>
<span><span class="co">#&gt;   Number of parameters: 4 </span></span>
<span><span class="co">#&gt;   Rhat:  1.004 (min), 1.04 (median), 1.074 (mean), 1.213 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 17 (min), 67.5 (median), 217 (mean), 716 (max) </span></span>
<span><span class="co">#&gt;  rho: </span></span>
<span><span class="co">#&gt;   Number of parameters: 6 </span></span>
<span><span class="co">#&gt;   Rhat:  1.042 (min), 1.107 (median), 1.106 (mean), 1.187 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 19 (min), 33.5 (median), 40.7 (mean), 79 (max) </span></span>
<span><span class="co">#&gt;  alpha: </span></span>
<span><span class="co">#&gt;   Number of parameters: 50 </span></span>
<span><span class="co">#&gt;   Rhat:  1.027 (min), 1.093 (median), 1.157 (mean), 1.77 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 7 (min), 34 (median), 40.6 (mean), 104 (max) </span></span>
<span><span class="co">#&gt;  beta: </span></span>
<span><span class="co">#&gt;   Number of parameters: 50 </span></span>
<span><span class="co">#&gt;   Rhat:  1.002 (min), 1.018 (median), 1.028 (mean), 1.237 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 16 (min), 183.5 (median), 349.9 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  gamma: </span></span>
<span><span class="co">#&gt;   Number of parameters: 100 </span></span>
<span><span class="co">#&gt;   Rhat:  1.001 (min), 1.019 (median), 1.029 (mean), 1.136 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 24 (min), 173.5 (median), 363.5 (mean), 2756 (max) </span></span>
<span><span class="co">#&gt;  alpha_shared: </span></span>
<span><span class="co">#&gt;   Number of parameters: 1 </span></span>
<span><span class="co">#&gt;   Rhat:  1.1 </span></span>
<span><span class="co">#&gt;   n.eff: 31 </span></span>
<span><span class="co">#&gt;  phi: </span></span>
<span><span class="co">#&gt;   Number of parameters: 50 </span></span>
<span><span class="co">#&gt;   Rhat:  1.001 (min), 1.07 (median), 1.127 (mean), 1.664 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 8 (min), 42.5 (median), 202.7 (mean), 3161 (max) </span></span>
<span><span class="co">#&gt;  theta: </span></span>
<span><span class="co">#&gt;   Number of parameters: 50 </span></span>
<span><span class="co">#&gt;   Rhat:  1.002 (min), 1.016 (median), 1.024 (mean), 1.219 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 16 (min), 196 (median), 358.1 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  psi: </span></span>
<span><span class="co">#&gt;   Number of parameters: 2500 </span></span>
<span><span class="co">#&gt;   Rhat:  1 (min), 1.008 (median), 1.013 (mean), 1.101 (max) </span></span>
<span><span class="co">#&gt;   n.eff: 36 (min), 396 (median), 679.3 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  z: </span></span>
<span><span class="co">#&gt;   Number of parameters: 2500 </span></span>
<span><span class="co">#&gt;   Rhat:  1 (min), 1.006 (median), 1.009 (mean), 1.184 (max), 946 (Number of NAs) </span></span>
<span><span class="co">#&gt;   n.eff: 1 (min), 1233.5 (median), 1628.6 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  pi: </span></span>
<span><span class="co">#&gt;   Number of parameters: 7500 </span></span>
<span><span class="co">#&gt;   Rhat:  1 (min), 1.078 (median), 1.102 (mean), 1.294 (max), 30 (Number of NAs) </span></span>
<span><span class="co">#&gt;   n.eff: 1 (min), 1 (median), 1009.1 (mean), 4000 (max) </span></span>
<span><span class="co">#&gt;  deviance: </span></span>
<span><span class="co">#&gt;   Rhat:  1.001 </span></span>
<span><span class="co">#&gt;   n.eff: 1332 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; **WARNING** Rhat values indicate convergence failure (Rhat &gt; 1.1). </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DIC info: (pD = var(deviance)/2)</span></span>
<span><span class="co">#&gt;  pD = 2131.5 and DIC = 19973.82 </span></span></code></pre></div>
<p>Overall, the model is not so far from convergence, but the
convergence of <code>phi</code> and <code>alpha</code>, the parameters
related to the relative dominance of species sequence reads, is not good
(i.e., high values of <code>Rhat</code>); hence, longer MCMC runs will
be required for a formal inference.</p>
<p>If you want to access posterior samples of individual parameters or
their summaries, you can use the <code><a href="../reference/get_posterior.html">get_post_samples()</a></code> and
<code><a href="../reference/get_posterior.html">get_post_summary()</a></code> functions. This may require
understanding the model parameters and their names: see
<code><a href="../articles/model_specification.html">vignette("model_specification")</a></code> for details. As an example,
try to get a posterior summary of site occupancy probability
<code>psi</code> using the <code><a href="../reference/get_posterior.html">get_post_summary()</a></code> function.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_summary_psi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_posterior.html">get_post_summary</a></span><span class="op">(</span><span class="va">fit1</span>, <span class="st">"psi"</span><span class="op">)</span></span>
<span><span class="va">post_summary_psi</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;                  mean          sd        2.5%        25%        50%        75%      97.5%     Rhat n.eff overlap0 f</span></span>
<span><span class="co">#&gt; psi[1,1]   0.64607375 0.083034133 0.466158634 0.59382744 0.65078275 0.70478301 0.79318634 1.001697  1264        0 1</span></span>
<span><span class="co">#&gt; psi[2,1]   0.28250539 0.089275014 0.136970706 0.21789079 0.27480943 0.33672122 0.47971573 1.013796   317        0 1</span></span>
<span><span class="co">#&gt; psi[3,1]   0.38321762 0.088583160 0.218908875 0.32150169 0.38055944 0.44171514 0.56510758 1.005127   472        0 1</span></span>
<span><span class="co">#&gt; psi[4,1]   0.16899696 0.070833671 0.058363836 0.11684541 0.15943020 0.21075828 0.32821669 1.001143  3788        0 1</span></span>
<span><span class="co">#&gt; psi[5,1]   0.20435372 0.088327893 0.069729103 0.13950376 0.19315615 0.25286705 0.41681236 1.006308   596        0 1</span></span>
<span><span class="co">#&gt; psi[6,1]   0.10737782 0.061179251 0.024299550 0.06246265 0.09573034 0.13925811 0.25577838 1.002291  1418        0 1</span></span>
<span><span class="co">#&gt; psi[7,1]   0.84639119 0.053478041 0.730770215 0.81327035 0.85048238 0.88494478 0.93751979 1.003575   654        0 1</span></span>
<span><span class="co">#&gt; psi[8,1]   0.94296846 0.028288919 0.875373936 0.92739641 0.94770976 0.96354354 0.98337253 1.002581  1006        0 1</span></span>
<span><span class="co">#&gt; (Omitted the remaining)</span></span></code></pre></div>
<p>In the <code>fit1</code> model, <code>psi</code> is assumed to have
different values for each site depending on the <code>riverbank</code>
covariate. Therefore, <code>psi</code> is estimated for each 50 species
<span class="math inline">\(\times\)</span> 50 sites (note that
<code>psi</code> differs for each species by default), and two
subscripts in parentheses distinguish them. So, which species and site
parameter is <code>psi[1,1]</code>? How about <code>psi[2,1]</code>?
Information about the dimension of the parameter is given in the
attributes of the resulting object.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_summary_psi</span><span class="op">)</span><span class="op">$</span><span class="va">dimension</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1] "Species" "Site"</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_summary_psi</span><span class="op">)</span><span class="op">$</span><span class="va">label</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; $Species</span></span>
<span><span class="co">#&gt;  [1] "Abbottina rivularis"                     </span></span>
<span><span class="co">#&gt;  [2] "Acanthogobius lactipes"                  </span></span>
<span><span class="co">#&gt;  [3] "Acheilognathus macropterus"              </span></span>
<span><span class="co">#&gt;  [4] "Acheilognathus rhombeus"                 </span></span>
<span><span class="co">#&gt;  [5] "Anguilla japonica"                       </span></span>
<span><span class="co">#&gt;  [6] "Biwia zezera"                            </span></span>
<span><span class="co">#&gt;  [7] "Carassius cuvieri"                       </span></span>
<span><span class="co">#&gt;  [8] "Carassius spp."                          </span></span>
<span><span class="co">#&gt;  [9] "Channa argus"                            </span></span>
<span><span class="co">#&gt; [10] "Ctenopharyngodon idella"                 </span></span>
<span><span class="co">#&gt; [11] "Cyprinus carpio"                         </span></span>
<span><span class="co">#&gt; [12] "Gambusia affinis"                        </span></span>
<span><span class="co">#&gt; [13] "Gnathopogon spp."                        </span></span>
<span><span class="co">#&gt; [14] "Gymnogobius castaneus"                   </span></span>
<span><span class="co">#&gt; [15] "Gymnogobius petschiliensis"              </span></span>
<span><span class="co">#&gt; [16] "Gymnogobius urotaenia"                   </span></span>
<span><span class="co">#&gt; [17] "Hemibarbus spp."                         </span></span>
<span><span class="co">#&gt; [18] "Hypomesus nipponensis"                   </span></span>
<span><span class="co">#&gt; [19] "Hypophthalmichthys spp."                 </span></span>
<span><span class="co">#&gt; [20] "Hyporhamphus intermedius"                </span></span>
<span><span class="co">#&gt; [21] "Ictalurus punctatus"                     </span></span>
<span><span class="co">#&gt; [22] "Ischikauia steenackeri"                  </span></span>
<span><span class="co">#&gt; [23] "Lepomis macrochirus macrochirus"         </span></span>
<span><span class="co">#&gt; [24] "Leucopsarion petersii"                   </span></span>
<span><span class="co">#&gt; [25] "Megalobrama amblycephala"                </span></span>
<span><span class="co">#&gt; [26] "Micropterus dolomieu dolomieu"           </span></span>
<span><span class="co">#&gt; [27] "Micropterus salmoides"                   </span></span>
<span><span class="co">#&gt; [28] "Misgurnus spp."                          </span></span>
<span><span class="co">#&gt; [29] "Monopterus albus"                        </span></span>
<span><span class="co">#&gt; [30] "Mugil cephalus cephalus"                 </span></span>
<span><span class="co">#&gt; [31] "Mylopharyngodon piceus"                  </span></span>
<span><span class="co">#&gt; [32] "Nipponocypris sieboldii"                 </span></span>
<span><span class="co">#&gt; [33] "Nipponocypris temminckii"                </span></span>
<span><span class="co">#&gt; [34] "Opsariichthys platypus"                  </span></span>
<span><span class="co">#&gt; [35] "Opsariichthys uncirostris uncirostris"   </span></span>
<span><span class="co">#&gt; [36] "Oryzias latipes"                         </span></span>
<span><span class="co">#&gt; [37] "Plecoglossus altivelis altivelis"        </span></span>
<span><span class="co">#&gt; [38] "Pseudogobio spp."                        </span></span>
<span><span class="co">#&gt; [39] "Pseudorasbora parva"                     </span></span>
<span><span class="co">#&gt; [40] "Rhinogobius spp."                        </span></span>
<span><span class="co">#&gt; [41] "Rhodeus ocellatus ocellatus"             </span></span>
<span><span class="co">#&gt; [42] "Salangichthys microdon"                  </span></span>
<span><span class="co">#&gt; [43] "Sarcocheilichthys variegatus microoculus"</span></span>
<span><span class="co">#&gt; [44] "Silurus asotus"                          </span></span>
<span><span class="co">#&gt; [45] "Squalidus chankaensis biwae"             </span></span>
<span><span class="co">#&gt; [46] "Tachysurus tokiensis"                    </span></span>
<span><span class="co">#&gt; [47] "Tanakia lanceolata"                      </span></span>
<span><span class="co">#&gt; [48] "Tribolodon brandtii maruta"              </span></span>
<span><span class="co">#&gt; [49] "Tribolodon hakonensis"                   </span></span>
<span><span class="co">#&gt; [50] "Tridentiger spp."                        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Site</span></span>
<span><span class="co">#&gt;  [1] "1"  "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9"  "10" "11" "12" "13" "14" "15"</span></span>
<span><span class="co">#&gt; [16] "16" "17" "18" "19" "20" "21" "22" "23" "24" "25" "26" "27" "28" "29" "30"</span></span>
<span><span class="co">#&gt; [31] "31" "32" "33" "34" "35" "36" "37" "38" "39" "40" "41" "42" "43" "44" "45"</span></span>
<span><span class="co">#&gt; [46] "46" "47" "48" "49" "50"</span></span></code></pre></div>
<p>Thus, from the <code>$dimension</code> attribute, we know that the
first subscript of <code>psi</code> represents the species and the
second the site. From the <code>$label</code> attribute, we know that
<code>psi[1,1]</code> represents the value of <code>psi</code> for
<em>Abbottina rivularis</em> at site 1, and <code>psi[2,1]</code>
represents the value of <code>psi</code> for <em>Acanthogobius
lactipes</em> at site 1.</p>
<p>We will also look at the posterior summary of the effect term
<code>gamma</code> on <code>psi</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_summary_gamma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_posterior.html">get_post_summary</a></span><span class="op">(</span><span class="va">fit1</span>, <span class="st">"gamma"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_summary_gamma</span><span class="op">)</span><span class="op">$</span><span class="va">dimension</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1] "Species" "Effects"</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_summary_gamma</span><span class="op">)</span><span class="op">$</span><span class="va">label</span><span class="op">$</span><span class="va">Effects</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1] "(Intercept)"                 "riverbankwithout_vegetation"</span></span></code></pre></div>
<p><code>gamma</code> has two effect values for each species: the
intercept <code>(Intercept)</code> and the effect of the absence of
vegetation in the riverbank <code>riverbankwithout_vegetation</code>.
Displaying the summary, we will see that, for many species, the absence
of vegetation has a negative effect on <code>psi</code> (on its link
scale).</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_summary_gamma</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;                    mean        sd       2.5%          25%         50%         75%        97.5%      Rhat n.eff overlap0       f</span></span>
<span><span class="co">#&gt; (Omitted the beginning)</span></span>
<span><span class="co">#&gt; gamma[1,2]  -0.68135951 0.4102181 -1.4106885 -0.964385346 -0.71198673 -0.41908665  0.197784615 1.0031831   720        1 0.93875</span></span>
<span><span class="co">#&gt; gamma[2,2]  -0.57124068 0.4861383 -1.4368696 -0.906331725 -0.60645174 -0.27472631  0.480190462 1.0141750   414        1 0.87925</span></span>
<span><span class="co">#&gt; gamma[3,2]  -1.07479024 0.4260032 -1.9550037 -1.336688784 -1.07308990 -0.80888005 -0.213626578 1.0040679   632        0 0.99250</span></span>
<span><span class="co">#&gt; gamma[4,2]  -0.96563859 0.5007208 -1.9760743 -1.276917710 -0.95208744 -0.64960250  0.016721077 1.0020286  1835        1 0.97225</span></span>
<span><span class="co">#&gt; gamma[5,2]  -1.01038376 0.5436033 -2.1340409 -1.339282006 -0.98783592 -0.65193007  0.006887964 1.0148153   190        1 0.97400</span></span>
<span><span class="co">#&gt; gamma[6,2]  -0.96443244 0.5694898 -2.1291123 -1.307706597 -0.96394933 -0.62099851  0.205483803 1.0043719   989        1 0.95325</span></span>
<span><span class="co">#&gt; gamma[7,2]  -1.44522243 0.4356167 -2.4011743 -1.721131931 -1.39965212 -1.14153735 -0.689367497 1.0021543  1065        0 0.99925</span></span>
<span><span class="co">#&gt; gamma[8,2]  -1.36146823 0.4820477 -2.3661876 -1.654265660 -1.32819584 -1.04709249 -0.483307820 1.0014346  1791        0 0.99900</span></span>
<span><span class="co">#&gt; (Omitted the remaining)</span></span></code></pre></div>
<p>Next, we will access posterior samples of <code>psi</code> using the
<code><a href="../reference/get_posterior.html">get_post_samples()</a></code> function.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_sample_psi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_posterior.html">get_post_samples</a></span><span class="op">(</span><span class="va">fit1</span>, <span class="st">"psi"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">post_sample_psi</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; [1] 4000   50   50</span></span></code></pre></div>
<p>The resulting object <code>post_sample_psi</code> is a 3-dimensional
array containing the posterior sample of <code>psi</code>. Information
about the dimensions of the extracted sample is given in its
attributes.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">post_sample_psi</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; $dim</span></span>
<span><span class="co">#&gt; [1] 4000   50   50</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $dimension</span></span>
<span><span class="co">#&gt; [1] "Sample"  "Species" "Site"   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $label</span></span>
<span><span class="co">#&gt; $label$Sample</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $label$Species</span></span>
<span><span class="co">#&gt;  [1] "Abbottina rivularis"                     </span></span>
<span><span class="co">#&gt;  [2] "Acanthogobius lactipes"                  </span></span>
<span><span class="co">#&gt;  [3] "Acheilognathus macropterus"              </span></span>
<span><span class="co">#&gt;  [4] "Acheilognathus rhombeus"                 </span></span>
<span><span class="co">#&gt;  [5] "Anguilla japonica"                       </span></span>
<span><span class="co">#&gt;  [6] "Biwia zezera"                            </span></span>
<span><span class="co">#&gt;  [7] "Carassius cuvieri"                       </span></span>
<span><span class="co">#&gt;  [8] "Carassius spp."                          </span></span>
<span><span class="co">#&gt;  [9] "Channa argus"                            </span></span>
<span><span class="co">#&gt; [10] "Ctenopharyngodon idella"                 </span></span>
<span><span class="co">#&gt; [11] "Cyprinus carpio"                         </span></span>
<span><span class="co">#&gt; [12] "Gambusia affinis"                        </span></span>
<span><span class="co">#&gt; [13] "Gnathopogon spp."                        </span></span>
<span><span class="co">#&gt; [14] "Gymnogobius castaneus"                   </span></span>
<span><span class="co">#&gt; [15] "Gymnogobius petschiliensis"              </span></span>
<span><span class="co">#&gt; [16] "Gymnogobius urotaenia"                   </span></span>
<span><span class="co">#&gt; [17] "Hemibarbus spp."                         </span></span>
<span><span class="co">#&gt; [18] "Hypomesus nipponensis"                   </span></span>
<span><span class="co">#&gt; [19] "Hypophthalmichthys spp."                 </span></span>
<span><span class="co">#&gt; [20] "Hyporhamphus intermedius"                </span></span>
<span><span class="co">#&gt; [21] "Ictalurus punctatus"                     </span></span>
<span><span class="co">#&gt; [22] "Ischikauia steenackeri"                  </span></span>
<span><span class="co">#&gt; [23] "Lepomis macrochirus macrochirus"         </span></span>
<span><span class="co">#&gt; [24] "Leucopsarion petersii"                   </span></span>
<span><span class="co">#&gt; [25] "Megalobrama amblycephala"                </span></span>
<span><span class="co">#&gt; [26] "Micropterus dolomieu dolomieu"           </span></span>
<span><span class="co">#&gt; [27] "Micropterus salmoides"                   </span></span>
<span><span class="co">#&gt; [28] "Misgurnus spp."                          </span></span>
<span><span class="co">#&gt; [29] "Monopterus albus"                        </span></span>
<span><span class="co">#&gt; [30] "Mugil cephalus cephalus"                 </span></span>
<span><span class="co">#&gt; [31] "Mylopharyngodon piceus"                  </span></span>
<span><span class="co">#&gt; [32] "Nipponocypris sieboldii"                 </span></span>
<span><span class="co">#&gt; [33] "Nipponocypris temminckii"                </span></span>
<span><span class="co">#&gt; [34] "Opsariichthys platypus"                  </span></span>
<span><span class="co">#&gt; [35] "Opsariichthys uncirostris uncirostris"   </span></span>
<span><span class="co">#&gt; [36] "Oryzias latipes"                         </span></span>
<span><span class="co">#&gt; [37] "Plecoglossus altivelis altivelis"        </span></span>
<span><span class="co">#&gt; [38] "Pseudogobio spp."                        </span></span>
<span><span class="co">#&gt; [39] "Pseudorasbora parva"                     </span></span>
<span><span class="co">#&gt; [40] "Rhinogobius spp."                        </span></span>
<span><span class="co">#&gt; [41] "Rhodeus ocellatus ocellatus"             </span></span>
<span><span class="co">#&gt; [42] "Salangichthys microdon"                  </span></span>
<span><span class="co">#&gt; [43] "Sarcocheilichthys variegatus microoculus"</span></span>
<span><span class="co">#&gt; [44] "Silurus asotus"                          </span></span>
<span><span class="co">#&gt; [45] "Squalidus chankaensis biwae"             </span></span>
<span><span class="co">#&gt; [46] "Tachysurus tokiensis"                    </span></span>
<span><span class="co">#&gt; [47] "Tanakia lanceolata"                      </span></span>
<span><span class="co">#&gt; [48] "Tribolodon brandtii maruta"              </span></span>
<span><span class="co">#&gt; [49] "Tribolodon hakonensis"                   </span></span>
<span><span class="co">#&gt; [50] "Tridentiger spp."                        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $label$Site</span></span>
<span><span class="co">#&gt;  [1] "1"  "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9"  "10" "11" "12" "13" "14" "15"</span></span>
<span><span class="co">#&gt; [16] "16" "17" "18" "19" "20" "21" "22" "23" "24" "25" "26" "27" "28" "29" "30"</span></span>
<span><span class="co">#&gt; [31] "31" "32" "33" "34" "35" "36" "37" "38" "39" "40" "41" "42" "43" "44" "45"</span></span>
<span><span class="co">#&gt; [46] "46" "47" "48" "49" "50"</span></span></code></pre></div>
<p>Thus, the second dimension of <code>post_sample_psi</code>
corresponds to the species and the third to the site. The labels for
species and sites indicate that <code>post_sample_psi[, 1, 1]</code>
contains the posterior sample of <code>psi</code> of <em>Abbottina
rivularis</em> at site 1.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_sample_psi</span><span class="op">[</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;  [1] 0.6673664 0.7505735 0.6416083 0.7011978 0.5935338 0.6087940 0.6159829</span></span>
<span><span class="co">#&gt;  [8] 0.7758206 0.6710466 0.7423708 0.6244308 0.6237291 0.6344345 0.6715390</span></span>
<span><span class="co">#&gt; [15] 0.6255393 0.7306071 0.6781538 0.7391794 0.6678208 0.6715868 0.7582302</span></span>
<span><span class="co">#&gt; [22] 0.5947020 0.4880676 0.6958859 0.6202520 0.5487584 0.6781225 0.7073072</span></span>
<span><span class="co">#&gt; (Omitted the remaining)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="assess-goodness-of-fit">Assess goodness-of-fit<a class="anchor" aria-label="anchor" href="#assess-goodness-of-fit"></a>
</h2>
<p>Assessing the goodness-of-fit of the fitted model is an essential
step in data analysis based on statistical models. The goodness-of-fit
of models fitted with <code><a href="../reference/occumb.html">occumb()</a></code> can be evaluated using the
<code><a href="../reference/gof.html">gof()</a></code> function which computes the Bayesian p-value via the
posterior predictive check approach.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gof_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gof.html">gof</a></span><span class="op">(</span><span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p>It is recommended to specify the <code>cores</code> argument
explicitly for parallel computation. By default, the <code><a href="../reference/gof.html">gof()</a></code>
function outputs a scatter plot of fit statistics (in this case, the
default Freeman-Tukey statistics).</p>
<p>To see a summary of the result, just enter the name of the resulting
object.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gof_result</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Posterior predictive check for an occumbFit object:</span></span>
<span><span class="co">#&gt;  Statistics: Freeman-Tukey </span></span>
<span><span class="co">#&gt;  p-value:    0.2005 </span></span>
<span><span class="co">#&gt;  Discrepancy statistics for observed data:   542.81 (mean), 17.92 (sd) </span></span>
<span><span class="co">#&gt;  Discrepancy statistics for replicated data: 523.28 (mean), 16.47 (sd) </span></span></code></pre></div>
<p>The moderate p-value indicates a lack of evidence of the model’s
inadequacy in fitting.</p>
<p>The <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> function can be used to (re)draw the scatter
plot of fit statistics from the resulting object.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gof_result</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/gof_fit1.png"></p>
</div>
<div class="section level2">
<h2 id="analyze-study-design">Analyze study design<a class="anchor" aria-label="anchor" href="#analyze-study-design"></a>
</h2>
<p>The model fitted with <code><a href="../reference/occumb.html">occumb()</a></code> can be used to find eDNA
metabarcoding study designs that effectively detect species present. How
many sites, within-site replicates, and sequencing depth will be
required to ensure reliable species detection? What is the best balance
between the number of sites visited, the number of within-site
replicates, and sequencing depth under a limited budget? The
multispecies site occupancy model fit by <code><a href="../reference/occumb.html">occumb()</a></code> answers
these questions by predicting the number of species expected to be
detected under specific study designs.</p>
<p><code><a href="../reference/eval_util_L.html">eval_util_L()</a></code> and <code><a href="../reference/eval_util_R.html">eval_util_R()</a></code> are
functions available for these purposes. These two functions assume
species diversity assessment at different spatial scales (i.e., the
former is for local (L), the latter for regional (R)). Specifically,
<code><a href="../reference/eval_util_L.html">eval_util_L()</a></code> is appropriate if you are interested in
assessing species diversity only for the study sites included in the
dataset, and <code><a href="../reference/eval_util_R.html">eval_util_R()</a></code> is appropriate if you are
interested in assessing species diversity in a broader area that
includes the study sites in the dataset (i.e., the population of sites
or a “metacommunity”).</p>
<p>In the context of the <code>fish</code> dataset,
<code><a href="../reference/eval_util_L.html">eval_util_L()</a></code> evaluates the expected number of species
detected per site at the 50 sites in the dataset under different
combinations of the number of replicates and sequencing depth. For
example, the following will give the expected number of species detected
per site (referred to as <code>Utility</code> below) when the number of
replicates <code>K</code> and the sequencing depth <code>N</code> take
the values (1, 2, 3) and (1000, 10000, 100000), respectively.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">utilL1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/eval_util_L.html">eval_util_L</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/expand.grid.html" class="external-link">expand.grid</a></span><span class="op">(</span>K <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, N <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1E3</span>, <span class="fl">1E4</span>, <span class="fl">1E5</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                      <span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilL1</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;   K     N  Utility</span></span>
<span><span class="co">#&gt; 1 1 1e+03 13.03225</span></span>
<span><span class="co">#&gt; 2 2 1e+03 16.72177</span></span>
<span><span class="co">#&gt; 3 3 1e+03 18.21783</span></span>
<span><span class="co">#&gt; 4 1 1e+04 14.34826</span></span>
<span><span class="co">#&gt; 5 2 1e+04 17.72304</span></span>
<span><span class="co">#&gt; 6 3 1e+04 18.92009</span></span>
<span><span class="co">#&gt; 7 1 1e+05 15.04591</span></span>
<span><span class="co">#&gt; 8 2 1e+05 18.20935</span></span>
<span><span class="co">#&gt; 9 3 1e+05 19.23478</span></span></code></pre></div>
<p>Note that it is recommended to specify the <code>cores</code>
argument explicitly for parallel computation.</p>
<p>If the cost per sequence read for high-throughput sequencing, the
cost per replicate for library preparation, and the research budget
values are each known, then the <code><a href="../reference/list_cond_L.html">list_cond_L()</a></code> function can
be used to obtain a set of feasible settings under these cost and budget
values. This can be given to <code><a href="../reference/eval_util_L.html">eval_util_L()</a></code> to identify the
optimal study design under a budget constraint.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">settings</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/list_cond_L.html">list_cond_L</a></span><span class="op">(</span>budget <span class="op">=</span> <span class="fl">875</span> <span class="op">*</span> <span class="fl">1E3</span>,</span>
<span>                        lambda1 <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                        lambda2 <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>                        <span class="va">fit1</span><span class="op">)</span></span>
<span><span class="va">utilL2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/eval_util_L.html">eval_util_L</a></span><span class="op">(</span><span class="va">settings</span>, <span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilL2</span></span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;   budget lambda1 lambda2 K          N  Utility</span></span>
<span><span class="co">#&gt; 1 875000    0.01    5000 1 1250000.00 15.45671</span></span>
<span><span class="co">#&gt; 2 875000    0.01    5000 2  375000.00 18.37862</span></span>
<span><span class="co">#&gt; 3 875000    0.01    5000 3   83333.33 19.21953</span></span></code></pre></div>
<p>Hence, under this specific budget and cost value, it is best to have
<code>K = 3</code> replicates per site for the most effective species
detection.</p>
<p>In contrast to <code><a href="../reference/eval_util_L.html">eval_util_L()</a></code>, <code><a href="../reference/eval_util_R.html">eval_util_R()</a></code>
considers differences in the number of sites visited. Nevertheless,
<code><a href="../reference/eval_util_R.html">eval_util_R()</a></code> can be applied in much the same way as
<code><a href="../reference/eval_util_L.html">eval_util_L()</a></code>. In <code><a href="../reference/eval_util_R.html">eval_util_R()</a></code>,
<code>Utility</code> is the number of species expected to be detected in
the region of interest under the settings of number of sites
<code>J</code>, number of replicates <code>K</code>, and the sequencing
depth <code>N</code>. Since <code><a href="../reference/list_cond_R.html">list_cond_R()</a></code> may return a huge
list of possible settings, you may want to get a manually restricted
list using its <code>J</code> and/or <code>K</code> arguments.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">utilR1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/eval_util_R.html">eval_util_R</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/expand.grid.html" class="external-link">expand.grid</a></span><span class="op">(</span>J <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, K <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, N <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1E3</span>, <span class="fl">1E4</span>, <span class="fl">1E5</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                      <span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilR1</span></span></code></pre></div>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; 1  1 1 1e+03 13.06227</span></span>
<span><span class="co">#&gt; 2  2 1 1e+03 18.77537</span></span>
<span><span class="co">#&gt; 3  3 1 1e+03 22.21901</span></span>
<span><span class="co">#&gt; 4  1 2 1e+03 16.75208</span></span>
<span><span class="co">#&gt; 5  2 2 1e+03 22.80510</span></span>
<span><span class="co">#&gt; 6  3 2 1e+03 26.33776</span></span>
<span><span class="co">#&gt; 7  1 3 1e+03 18.18526</span></span>
<span><span class="co">#&gt; 8  2 3 1e+03 24.63089</span></span>
<span><span class="co">#&gt; 9  3 3 1e+03 28.13433</span></span>
<span><span class="co">#&gt; 10 1 1 1e+04 14.36287</span></span>
<span><span class="co">#&gt; 11 2 1 1e+04 20.30642</span></span>
<span><span class="co">#&gt; 12 3 1 1e+04 23.85725</span></span>
<span><span class="co">#&gt; 13 1 2 1e+04 17.65318</span></span>
<span><span class="co">#&gt; 14 2 2 1e+04 23.98433</span></span>
<span><span class="co">#&gt; 15 3 2 1e+04 27.55905</span></span>
<span><span class="co">#&gt; 16 1 3 1e+04 19.03370</span></span>
<span><span class="co">#&gt; 17 2 3 1e+04 25.30884</span></span>
<span><span class="co">#&gt; 18 3 3 1e+04 29.14037</span></span>
<span><span class="co">#&gt; 19 1 1 1e+05 14.97290</span></span>
<span><span class="co">#&gt; 20 2 1 1e+05 21.16216</span></span>
<span><span class="co">#&gt; 21 3 1 1e+05 24.76543</span></span>
<span><span class="co">#&gt; 22 1 2 1e+05 18.25193</span></span>
<span><span class="co">#&gt; 23 2 2 1e+05 24.64478</span></span>
<span><span class="co">#&gt; 24 3 2 1e+05 28.20150</span></span>
<span><span class="co">#&gt; 25 1 3 1e+05 19.33757</span></span>
<span><span class="co">#&gt; 26 2 3 1e+05 25.85813</span></span>
<span><span class="co">#&gt; 27 3 3 1e+05 29.68874</span></span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">settings</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/list_cond_R.html">list_cond_R</a></span><span class="op">(</span>budget <span class="op">=</span> <span class="fl">1125</span> <span class="op">*</span> <span class="fl">1E3</span>,</span>
<span>                        lambda1 <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                        lambda2 <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>                        lambda3 <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>                        J <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">50</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>                        K <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilR2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/eval_util_R.html">eval_util_R</a></span><span class="op">(</span><span class="va">settings</span>, <span class="va">fit1</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">utilR2</span></span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;     budget lambda1 lambda2 lambda3  J K           N  Utility</span></span>
<span><span class="co">#&gt; 1  1125000    0.01    5000    5000  5 1 21500000.00 30.04136</span></span>
<span><span class="co">#&gt; 2  1125000    0.01    5000    5000 10 1 10250000.00 35.64544</span></span>
<span><span class="co">#&gt; 3  1125000    0.01    5000    5000 15 1  6500000.00 38.60797</span></span>
<span><span class="co">#&gt; 4  1125000    0.01    5000    5000 20 1  4625000.00 40.60944</span></span>
<span><span class="co">#&gt; 5  1125000    0.01    5000    5000 25 1  3500000.00 42.02112</span></span>
<span><span class="co">#&gt; 6  1125000    0.01    5000    5000 30 1  2750000.00 43.15096</span></span>
<span><span class="co">#&gt; 7  1125000    0.01    5000    5000 35 1  2214285.71 43.89833</span></span>
<span><span class="co">#&gt; 8  1125000    0.01    5000    5000 40 1  1812500.00 44.71556</span></span>
<span><span class="co">#&gt; 9  1125000    0.01    5000    5000 45 1  1500000.00 45.22666</span></span>
<span><span class="co">#&gt; 10 1125000    0.01    5000    5000 50 1  1250000.00 45.71167</span></span>
<span><span class="co">#&gt; 11 1125000    0.01    5000    5000  5 2 10500000.00 33.34409</span></span>
<span><span class="co">#&gt; 12 1125000    0.01    5000    5000 10 2  4875000.00 38.81906</span></span>
<span><span class="co">#&gt; 13 1125000    0.01    5000    5000 15 2  3000000.00 41.58322</span></span>
<span><span class="co">#&gt; 14 1125000    0.01    5000    5000 20 2  2062500.00 43.47530</span></span>
<span><span class="co">#&gt; 15 1125000    0.01    5000    5000 25 2  1500000.00 44.69525</span></span>
<span><span class="co">#&gt; 16 1125000    0.01    5000    5000 30 2  1125000.00 45.58398</span></span>
<span><span class="co">#&gt; 17 1125000    0.01    5000    5000 35 2   857142.86 46.27036</span></span>
<span><span class="co">#&gt; 18 1125000    0.01    5000    5000 40 2   656250.00 46.82490</span></span>
<span><span class="co">#&gt; 19 1125000    0.01    5000    5000 45 2   500000.00 47.24841</span></span>
<span><span class="co">#&gt; 20 1125000    0.01    5000    5000 50 2   375000.00 47.56585</span></span>
<span><span class="co">#&gt; 21 1125000    0.01    5000    5000  5 3  6833333.33 34.55557</span></span>
<span><span class="co">#&gt; 22 1125000    0.01    5000    5000 10 3  3083333.33 40.03808</span></span>
<span><span class="co">#&gt; 23 1125000    0.01    5000    5000 15 3  1833333.33 42.85560</span></span>
<span><span class="co">#&gt; 24 1125000    0.01    5000    5000 20 3  1208333.33 44.51759</span></span>
<span><span class="co">#&gt; 25 1125000    0.01    5000    5000 25 3   833333.33 45.67757</span></span>
<span><span class="co">#&gt; 26 1125000    0.01    5000    5000 30 3   583333.33 46.48685</span></span>
<span><span class="co">#&gt; 27 1125000    0.01    5000    5000 35 3   404761.90 47.14072</span></span>
<span><span class="co">#&gt; 28 1125000    0.01    5000    5000 40 3   270833.33 47.60570</span></span>
<span><span class="co">#&gt; 29 1125000    0.01    5000    5000 45 3   166666.67 47.93433</span></span>
<span><span class="co">#&gt; 30 1125000    0.01    5000    5000 50 3    83333.33 48.12801</span></span>
<span><span class="co">#&gt; 31 1125000    0.01    5000    5000  5 4  5000000.00 35.17767</span></span>
<span><span class="co">#&gt; 32 1125000    0.01    5000    5000 10 4  2187500.00 40.72769</span></span>
<span><span class="co">#&gt; 33 1125000    0.01    5000    5000 15 4  1250000.00 43.51095</span></span>
<span><span class="co">#&gt; 34 1125000    0.01    5000    5000 20 4   781250.00 45.16898</span></span>
<span><span class="co">#&gt; 35 1125000    0.01    5000    5000 25 4   500000.00 46.23139</span></span>
<span><span class="co">#&gt; 36 1125000    0.01    5000    5000 30 4   312500.00 46.98586</span></span>
<span><span class="co">#&gt; 37 1125000    0.01    5000    5000 35 4   178571.43 47.53393</span></span>
<span><span class="co">#&gt; 38 1125000    0.01    5000    5000 40 4    78125.00 47.89744</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Keiichi Fukaya.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
